// @opentiny/tiny-robot@0.3.0-alpha.14/dist/container/index.js
import { defineComponent as v, useCssVars as m, useModel as d, computed as _, createElementBlock as w, openBlock as g, createElementVNode as o, renderSlot as r, createVNode as u, unref as a } from "vue";
import { m as h, G as $, x as C } from "../tiny-robot-svgs.js";
import c from "../icon-button/index.js";
import { _ as b } from "../_plugin-vue_export-helper.js";
const z = { class: "tr-container" }, B = { class: "tr-container__header" }, k = { class: "tr-container__header-operations" }, x = { class: "tr-container__footer" }, y = /* @__PURE__ */ v({
  __name: "index",
  props: {
    show: { type: Boolean, required: !0 },
    showModifiers: {},
    fullscreen: { type: Boolean },
    fullscreenModifiers: {}
  },
  emits: ["update:show", "update:fullscreen"],
  setup(n) {
    m((t) => ({
      "12c6880d": s.value ? "0" : "unset",
      dbe5f3c0: s.value ? "unset" : "var(--tr-container-width)",
      "676af261": l.value ? "var(--tr-z-index-fixed)" : "-1",
      "19e9bbe2": l.value ? "1" : "0"
    }));
    const l = d(n, "show"), s = d(n, "fullscreen"), f = _(() => s.value ? h : $);
    return (t, e) => (g(), w("div", z, [
      e[3] || (e[3] = o("div", { class: "tr-container__dragging-bar-wrapper" }, [
        o("div", { class: "tr-container__dragging-bar" })
      ], -1)),
      o("div", B, [
        r(t.$slots, "title", {}, () => [
          e[2] || (e[2] = o("h3", { class: "tr-container__title" }, "OpenTiny NEXT", -1))
        ], !0),
        o("div", k, [
          r(t.$slots, "operations", {}, void 0, !0),
          u(a(c), {
            size: "28",
            "svg-size": "20",
            icon: f.value,
            onClick: e[0] || (e[0] = (p) => t.$emit("update:fullscreen", !s.value))
          }, null, 8, ["icon"]),
          u(a(c), {
            size: "28",
            "svg-size": "20",
            icon: a(C),
            onClick: e[1] || (e[1] = (p) => t.$emit("update:show", !1))
          }, null, 8, ["icon"])
        ])
      ]),
      r(t.$slots, "default", {}, void 0, !0),
      o("div", x, [
        r(t.$slots, "footer", {}, void 0, !0)
      ])
    ]));
  }
}), i = /* @__PURE__ */ b(y, [["__scopeId", "data-v-3fa99833"]]);
i.name = "TrContainer";
const E = function(n) {
  n.component(i.name, i);
};
i.install = E;
export {
  i as default
};
