// @opentiny/tiny-robot@0.3.0-alpha.14/dist/dropdown-menu/index.js
import { ref as k, onUnmounted as T, watch as C, defineComponent as E, mergeModels as y, useModel as H, computed as p, createBlock as I, openBlock as w, unref as X, withCtx as _, createElementVNode as j, createElementBlock as b, Fragment as D, renderList as L, toDisplayString as Y, renderSlot as O } from "vue";
import { w as S, b as M, o as $ } from "../index3.js";
import { B as A } from "../index4.js";
import { u as F } from "../utils.js";
import { _ as G } from "../_plugin-vue_export-helper.js";
const P = (o, t) => o.x >= t.left && o.x <= t.right && o.y >= t.top && o.y <= t.bottom, N = (o, t, g = !1) => {
  if (t.length < 3) return !1;
  const l = o.x, d = o.y;
  let f = !1;
  const r = (s, c, a, n, i, e) => {
    const m = (i - a) * (c - n) - (e - n) * (s - a);
    if (Math.abs(m) > 1e-10) return !1;
    const u = Math.min(a, i), h = Math.max(a, i), x = Math.min(n, e), B = Math.max(n, e);
    return s >= u && s <= h && c >= x && c <= B;
  };
  for (let s = 0, c = t.length - 1; s < t.length; c = s++) {
    const a = t[s].x, n = t[s].y, i = t[c].x, e = t[c].y;
    if (g && r(l, d, a, n, i, e))
      return !0;
    if (n > d && e <= d || e > d && n <= d) {
      const u = (i - a) * (d - n) / (e - n) + a;
      l < u && (f = !f);
    }
  }
  return f;
}, U = (o, t) => t.bottom < o.top ? [
  { x: o.left, y: o.top },
  { x: o.right, y: o.top },
  { x: t.right, y: t.bottom },
  { x: t.left, y: t.bottom }
] : t.top > o.bottom ? [
  { x: o.left, y: o.bottom },
  { x: o.right, y: o.bottom },
  { x: t.right, y: t.top },
  { x: t.left, y: t.top }
] : [], { x: V, y: q } = F(), z = (o, t, g) => {
  const { delayEnter: l = 0, delayLeave: d = 0 } = g || {}, f = k(!1), r = k("outside");
  let s = null;
  const c = () => {
    s && (clearTimeout(s), s = null);
  };
  T(() => {
    c();
  });
  const a = (n) => {
    const i = n ? l : d;
    c(), i ? s = setTimeout(() => f.value = n, i) : f.value = n;
  };
  return C(
    r,
    (n) => {
      ["trigger", "popper", "polygon"].includes(n) ? a(!0) : a(!1);
    },
    { immediate: !0 }
  ), S(
    [V, q],
    ([n, i]) => {
      const e = M(o), m = M(t), u = e == null ? void 0 : e.getBoundingClientRect(), h = m == null ? void 0 : m.getBoundingClientRect();
      if (u && P({ x: n, y: i }, u)) {
        r.value = "trigger";
        return;
      }
      if (h && P({ x: n, y: i }, h)) {
        r.value = "popper";
        return;
      }
      if (!u || !h || u.width === 0 || u.height === 0 || h.width === 0 || h.height === 0) {
        r.value = "outside";
        return;
      }
      N({ x: n, y: i }, U(u, h), !0) ? r.value = "polygon" : r.value = "outside";
    },
    { throttle: 10 }
  ), f;
}, J = { class: "tr-dropdown-menu__list" }, K = ["onClick"], Q = /* @__PURE__ */ E({
  __name: "index",
  props: /* @__PURE__ */ y({
    appendTo: {},
    items: {},
    show: { type: Boolean },
    trigger: { default: "click" }
  }, {
    show: { type: Boolean, default: !1 },
    showModifiers: {}
  }),
  emits: /* @__PURE__ */ y(["item-click", "click-outside"], ["update:show"]),
  setup(o, { expose: t, emit: g }) {
    const l = o, d = g, f = H(o, "show"), r = p({
      get: () => l.trigger === "manual" ? l.show : f.value,
      set: (e) => {
        l.trigger !== "manual" && (f.value = e);
      }
    }), s = k(null), c = p(() => {
      var e;
      return (e = s.value) == null ? void 0 : e.triggerRef;
    }), a = p(() => {
      var e;
      return (e = s.value) == null ? void 0 : e.popperRef;
    });
    if (l.trigger === "click" || l.trigger === "manual")
      $(
        a,
        (e) => {
          d("click-outside", e), r.value = !1;
        },
        { ignore: [c] }
      );
    else if (l.trigger === "hover") {
      const e = z(c, a, { delayEnter: 100, delayLeave: 100 });
      C(e, (m) => {
        r.value = m;
      });
    }
    const n = () => {
      l.trigger === "click" && (r.value = !r.value);
    }, i = (e) => {
      r.value = !1, d("item-click", e);
    };
    return t({
      update: () => {
        var e;
        (e = s.value) == null || e.update();
      }
    }), (e, m) => (w(), I(X(A), {
      show: r.value,
      class: "tr-dropdown-menu",
      ref_key: "basePopperRef",
      ref: s,
      placement: "top-left",
      "append-to": l.appendTo,
      offset: 8,
      "transition-props": { name: "tr-dropdown-menu" },
      "prevent-overflow": !0,
      "trigger-events": { onClick: n }
    }, {
      trigger: _(() => [
        O(e.$slots, "trigger", {}, void 0, !0)
      ]),
      content: _(() => [
        j("ul", J, [
          (w(!0), b(D, null, L(l.items, (u) => (w(), b("li", {
            class: "tr-dropdown-menu__list-item",
            key: u.id,
            onClick: (h) => i(u)
          }, Y(u.text), 9, K))), 128))
        ])
      ]),
      _: 3
    }, 8, ["show", "append-to", "trigger-events"]));
  }
}), v = /* @__PURE__ */ G(Q, [["__scopeId", "data-v-8d480727"]]);
v.name = "TrDropdownMenu";
const W = function(o) {
  o.component(v.name, v);
};
v.install = W;
export {
  v as default
};
