// @opentiny/tiny-engine@2.7.2/dist/index.js
import "./style.css";
import { e as t, i as o, d as f } from "./index-D_kziuWO.js";
import { default as s, BreadcrumbService as u } from "@opentiny/tiny-engine-toolbar-breadcrumb";
import { default as d } from "@opentiny/tiny-engine-toolbar-fullscreen";
import { default as m } from "@opentiny/tiny-engine-toolbar-lang";
import { default as c } from "@opentiny/tiny-engine-toolbar-view-setting";
import { default as v } from "@opentiny/tiny-engine-toolbar-logo";
import { default as g } from "@opentiny/tiny-engine-toolbar-lock";
import { default as b } from "@opentiny/tiny-engine-toolbar-media";
import { HistoryService as y, default as B } from "@opentiny/tiny-engine-toolbar-redoundo";
import { default as w } from "@opentiny/tiny-engine-toolbar-save";
import { default as R } from "@opentiny/tiny-engine-toolbar-clean";
import { default as H, ThemeSwitchService as k } from "@opentiny/tiny-engine-toolbar-theme-switch";
import { default as M } from "@opentiny/tiny-engine-toolbar-preview";
import { default as E, SaveLocalService as F } from "@opentiny/tiny-engine-toolbar-generate-code";
import { default as V } from "@opentiny/tiny-engine-toolbar-refresh";
import { default as q } from "@opentiny/tiny-engine-toolbar-collaboration";
import { default as A } from "@opentiny/tiny-engine-toolbar-setting";
import { MaterialService as K, default as N, ResourceService as O } from "@opentiny/tiny-engine-plugin-materials";
import { default as U } from "@opentiny/tiny-engine-plugin-state";
import { default as X } from "@opentiny/tiny-engine-plugin-script";
import { default as Z } from "@opentiny/tiny-engine-plugin-tree";
import { default as $, HelpService as ee } from "@opentiny/tiny-engine-plugin-help";
import { default as ae } from "@opentiny/tiny-engine-plugin-schema";
import { default as oe, PageService as fe } from "@opentiny/tiny-engine-plugin-page";
import { default as se, TranslateService as ue } from "@opentiny/tiny-engine-plugin-i18n";
import { default as de } from "@opentiny/tiny-engine-plugin-bridge";
import { default as me, BlockService as xe } from "@opentiny/tiny-engine-plugin-block";
import { DataSourceService as Se, default as ve } from "@opentiny/tiny-engine-plugin-datasource";
import { default as ge } from "@opentiny/tiny-engine-plugin-robot";
import { PropertiesService as be, PropertyService as he, default as ye } from "@opentiny/tiny-engine-setting-props";
import { default as Le } from "@opentiny/tiny-engine-setting-events";
import { default as Ce } from "@opentiny/tiny-engine-setting-styles";
import { default as Te, LayoutService as He } from "@opentiny/tiny-engine-layout";
import { default as Ge } from "@opentiny/tiny-engine-canvas";
import { GenerateCodeService as De, GlobalService as Ee, HttpService as Fe, PluginPanel as Ie, PluginSetting as Ve, ToolbarBase as je } from "@opentiny/tiny-engine-common";
export * from "@opentiny/tiny-engine-meta-register";
export {
  me as Block,
  xe as BlockService,
  s as Breadcrumb,
  u as BreadcrumbService,
  de as Bridge,
  Ge as Canvas,
  R as Clean,
  q as Collaboration,
  Se as DataSourceService,
  ve as Datasource,
  Le as Events,
  d as Fullscreen,
  E as GenerateCode,
  De as GenerateCodeService,
  Ee as GlobalService,
  $ as Help,
  ee as HelpService,
  y as HistoryService,
  Fe as HttpService,
  se as I18n,
  m as Lang,
  Te as Layout,
  He as LayoutService,
  g as Lock,
  v as Logo,
  K as MaterialService,
  N as Materials,
  b as Media,
  oe as Page,
  fe as PageService,
  Ie as PluginPanel,
  Ve as PluginSetting,
  M as Preview,
  be as PropertiesService,
  he as PropertyService,
  ye as Props,
  B as Redoundo,
  V as Refresh,
  O as ResourceService,
  ge as Robot,
  w as Save,
  F as SaveLocalService,
  ae as Schema,
  X as Script,
  A as Setting,
  U as State,
  Ce as Styles,
  H as ThemeSwitch,
  k as ThemeSwitchService,
  je as ToolbarBase,
  ue as TranslateService,
  Z as Tree,
  c as ViewSetting,
  t as defaultRegistry,
  o as init,
  f as initPreview
};
