// @opentiny/tiny-robot@0.3.0-alpha.14/dist/suggestion-pills/index.js
import { defineComponent as K, useSlots as Q, computed as a, createElementBlock as m, openBlock as r, normalizeClass as w, renderSlot as H, createBlock as S, resolveDynamicComponent as b, createCommentVNode as M, toDisplayString as ne, mergeModels as N, useModel as le, ref as d, watch as R, nextTick as $, createElementVNode as D, Fragment as z, renderList as G, isVNode as j, createVNode as q, Transition as se, withCtx as re, unref as ae } from "vue";
import { g as ie } from "../tiny-robot-svgs.js";
import { e as ce, b as J, f as ue, o as fe, g as de } from "../index3.js";
import { u as me } from "../useSlotRefs.js";
import { _ as U } from "../_plugin-vue_export-helper.js";
const pe = { key: 0 }, ge = /* @__PURE__ */ K({
  __name: "PillButton",
  props: {
    item: {}
  },
  setup(i) {
    const v = i, _ = Q(), p = a(() => {
      var o;
      return !!(_.icon || (o = v.item) != null && o.icon);
    }), I = a(() => {
      var o;
      return !!(_.default || (o = v.item) != null && o.text);
    }), x = a(() => p.value && !I.value);
    return (o, u) => (r(), m("button", {
      class: w(["tr-suggestion-pills__item", { "only-icon": x.value }])
    }, [
      H(o.$slots, "icon", {}, () => {
        var c;
        return [
          (r(), S(b((c = o.item) == null ? void 0 : c.icon), { class: "tr-suggestion-pills__item_icon" }))
        ];
      }, !0),
      H(o.$slots, "default", {}, () => {
        var c;
        return [
          (c = o.item) != null && c.text ? (r(), m("span", pe, ne(o.item.text), 1)) : M("", !0)
        ];
      }, !0)
    ], 2));
  }
}), h = /* @__PURE__ */ U(ge, [["__scopeId", "data-v-5c433c31"]]), he = { class: "tr-suggestion-pills__more-wrapper" }, ve = /* @__PURE__ */ K({
  __name: "index",
  props: /* @__PURE__ */ N({
    showAll: { type: Boolean },
    showAllButtonOn: { default: "hover" },
    overflowMode: { default: "expand" },
    autoScrollOn: {}
  }, {
    showAll: { type: Boolean, default: !1 },
    showAllModifiers: {}
  }),
  emits: /* @__PURE__ */ N(["click-outside"], ["update:showAll"]),
  setup(i, { expose: v, emit: _ }) {
    const p = i, I = Q(), x = _, { vnodes: o } = me(I.default, !0), u = le(i, "showAll"), c = d(null), f = d(null), A = d(null), { width: W } = ce(f), C = d(0), B = a(() => p.overflowMode === "expand" && W.value < C.value), g = d(-1), X = a(() => B.value && u.value ? o.value.slice(0, g.value) : o.value), T = a(() => B.value && u.value ? o.value.slice(g.value).map((e, t) => ({
      vnode: e,
      index: t + g.value
    })) : []), E = d([]), P = d([]), V = a(() => E.value.map((e) => J(e)).filter((e) => e instanceof Element)), Y = a(() => P.value.map((e) => J(e)).filter((e) => e instanceof Element)), L = () => {
      const e = f.value, t = A.value;
      return Array.from((e == null ? void 0 : e.children) || []).concat(Array.from((t == null ? void 0 : t.children) || []));
    }, O = () => {
      $(() => {
        const e = f.value;
        if (!e)
          return;
        const t = L(), n = parseFloat(getComputedStyle(e).rowGap) || 0;
        g.value = -1;
        let l = 0;
        for (let s = 0; s < t.length; s++)
          if (l += t[s].offsetWidth, s > 0 && (l += n), l > e.clientWidth) {
            g.value = s;
            break;
          }
      });
    };
    R(
      o,
      () => {
        $(() => {
          if (!f.value)
            return;
          const e = L(), t = parseFloat(getComputedStyle(f.value).rowGap) || 0;
          C.value = e.reduce((n, l, s) => n + l.offsetWidth + (s > 0 ? t : 0), 0);
        });
      },
      { immediate: !0 }
    ), R(o, O), ue(W, O, { debounce: 100 });
    const Z = () => {
      u.value = !u.value;
    };
    fe(c, (e) => {
      u.value && x("click-outside", e);
    });
    const ee = (e) => {
      const t = f.value;
      if (!t)
        return;
      const n = e.offsetLeft, l = n + e.offsetWidth, s = t.scrollLeft, F = t.clientWidth, te = n < s, oe = l > s + F;
      te ? t.scrollTo({
        left: n,
        behavior: "smooth"
      }) : oe && t.scrollTo({
        left: l - F,
        behavior: "smooth"
      });
    };
    let y = null;
    return R(
      () => p.autoScrollOn,
      (e) => {
        y && (y(), y = null), e && (y = de(V, e, (t) => {
          t.currentTarget && ee(t.currentTarget);
        }));
      },
      {
        immediate: !0
      }
    ), v({
      children: a(() => V.value.concat(Y.value))
    }), (e, t) => (r(), m("div", {
      class: "tr-suggestion-pills__wrapper",
      ref_key: "containerWrapperRef",
      ref: c
    }, [
      D("div", {
        class: w(["tr-suggestion-pills__container", { "overflow-scroll": p.overflowMode === "scroll" }]),
        ref_key: "containerRef",
        ref: f
      }, [
        (r(!0), m(z, null, G(X.value, (n, l) => (r(), S(b(n), {
          key: j(n) ? n.key : l,
          ref_for: !0,
          ref_key: "staticMaybeItemRefs",
          ref: E
        }))), 128))
      ], 2),
      D("div", he, [
        q(se, { name: "tr-suggestion-pills__more" }, {
          default: re(() => [
            T.value.length ? (r(), m("div", {
              key: 0,
              class: "tr-suggestion-pills__more",
              ref_key: "floatingItemsRef",
              ref: A
            }, [
              (r(!0), m(z, null, G(T.value, ({ vnode: n, index: l }) => (r(), S(b(n), {
                key: j(n) ? n.key : l,
                ref_for: !0,
                ref_key: "floatingMaybeItemRefs",
                ref: P
              }))), 128))
            ], 512)) : M("", !0)
          ]),
          _: 1
        })
      ]),
      B.value ? (r(), m("button", {
        key: 0,
        class: w(["tr-suggestion-pills__expand", { "show-on-hover": p.showAllButtonOn === "hover" }]),
        onClick: Z
      }, [
        q(ae(ie), {
          class: w(["tr-suggestion-pills__expand-icon", { rotate: u.value }])
        }, null, 8, ["class"])
      ], 2)) : M("", !0)
    ], 512));
  }
}), k = /* @__PURE__ */ U(ve, [["__scopeId", "data-v-ebbd7a00"]]);
k.name = "TrSuggestionPills";
const _e = function(i) {
  i.component(k.name, k);
};
k.install = _e;
h.name = "TrSuggestionPillButton";
const ye = function(i) {
  i.component(h.name, h);
};
h.install = ye;
const Re = h;
export {
  Re as SuggestionPillButton,
  k as default
};
