// @opentiny/tiny-robot@0.3.0-alpha.14/dist/tiny-robot-svgs.js
import { createElementBlock as r, openBlock as C, createStaticVNode as h, createElementVNode as i } from "vue";
const d = (t, l) => {
  const e = t.__vccOpts || t;
  for (const [o, n] of l)
    e[o] = n;
  return e;
}, c = {
  name: "IconArrowDown"
}, p = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 16 16",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function a(t, l, e, o, n, s) {
  return C(), r("svg", p, l[0] || (l[0] = [
    i("defs", null, null, -1),
    i("path", {
      id: "路径",
      d: "M 3.8 6 L 8 10.2 L 12.2 6",
      stroke: "currentColor",
      "stroke-width": "1",
      "stroke-linejoin": "round",
      "stroke-linecap": "round"
    }, null, -1)
  ]));
}
const y0 = /* @__PURE__ */ d(c, [["render", a]]), w = {
  name: "IconArrowLeft"
}, L = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 16 16",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function f(t, l, e, o, n, s) {
  return C(), r("svg", L, l[0] || (l[0] = [
    i("defs", null, null, -1),
    i("path", {
      id: "路径",
      d: "M 10 3.8 L 5.8 8 L 10 12.2",
      stroke: "currentColor",
      "stroke-width": "1",
      "stroke-linejoin": "round",
      "stroke-linecap": "round"
    }, null, -1)
  ]));
}
const M0 = /* @__PURE__ */ d(w, [["render", f]]), u = {
  name: "IconArrowRight"
}, g = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 16 16",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function x(t, l, e, o, n, s) {
  return C(), r("svg", g, l[0] || (l[0] = [
    i("defs", null, null, -1),
    i("path", {
      id: "路径",
      d: "M 6 3.8 L 10.2 8 L 6 12.2",
      stroke: "currentColor",
      "stroke-width": "1",
      "stroke-linejoin": "round",
      "stroke-linecap": "round"
    }, null, -1)
  ]));
}
const Z0 = /* @__PURE__ */ d(u, [["render", x]]), k = {
  name: "IconArrowUp"
}, F = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 16 16",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function m(t, l, e, o, n, s) {
  return C(), r("svg", F, l[0] || (l[0] = [
    i("defs", null, null, -1),
    i("path", {
      id: "路径",
      d: "M 3.8 10 L 8 5.8 L 12.2 10",
      stroke: "currentColor",
      "stroke-width": "1",
      "stroke-linejoin": "round",
      "stroke-linecap": "round"
    }, null, -1)
  ]));
}
const _0 = /* @__PURE__ */ d(k, [["render", m]]), v = {
  name: "IconAssociate"
}, y = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 16 16",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function M(t, l, e, o, n, s) {
  return C(), r("svg", y, l[0] || (l[0] = [
    i("rect", {
      id: "矩形",
      width: "16.000000",
      height: "16.000000",
      fill: "#D8D8D8",
      "fill-opacity": "0"
    }, null, -1),
    i("rect", {
      id: "矩形",
      x: "0.500000",
      y: "0.500000",
      width: "15.000000",
      height: "15.000000",
      stroke: "#979797",
      "stroke-opacity": "0",
      "stroke-width": "1.000000"
    }, null, -1),
    i("path", {
      id: "Fill 1",
      d: "M9.79 6.23C7.34 6.66 6.66 7.34 6.23 9.79C6.19 10.06 5.8 10.06 5.76 9.79C5.33 7.34 4.65 6.66 2.2 6.23C1.93 6.19 1.93 5.8 2.2 5.76C4.65 5.33 5.33 4.65 5.76 2.2C5.8 1.93 6.19 1.93 6.23 2.2C6.66 4.65 7.34 5.33 9.79 5.76C10.06 5.8 10.06 6.19 9.79 6.23Z",
      fill: "#1476FF",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1),
    i("path", {
      id: "Fill 1备份",
      d: "M13.87 11.64C12.33 11.91 11.91 12.33 11.64 13.87C11.62 14.04 11.37 14.04 11.35 13.87C11.08 12.33 10.66 11.91 9.12 11.64C8.95 11.62 8.95 11.37 9.12 11.35C10.66 11.08 11.08 10.66 11.35 9.12C11.37 8.95 11.62 8.95 11.64 9.12C11.91 10.66 12.33 11.08 13.87 11.35C14.04 11.37 14.04 11.62 13.87 11.64Z",
      fill: "#1476FF",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1)
  ]));
}
const B0 = /* @__PURE__ */ d(v, [["render", M]]), Z = {
  name: "IconCancelFullScreen"
}, _ = {
  width: "1em",
  height: "1em",
  version: "1.1",
  id: "图层_1",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink",
  x: "0px",
  y: "0px",
  viewBox: "0 0 132 132",
  style: { "enable-background": "new 0 0 132 132" },
  "xml:space": "preserve"
};
function B(t, l, e, o, n, s) {
  return C(), r("svg", _, l[0] || (l[0] = [
    i("rect", {
      style: { fill: "none" },
      width: "132",
      height: "132"
    }, null, -1),
    i("path", {
      style: { fill: "currentColor" },
      d: `M96.2,64.8c2.3,0,4.1-1.8,4.1-4.1s-1.8-4.1-4.1-4.1H81.3l20.6-20.7c1.5-1.4,1.6-3.7,0.4-5.4l-0.4-0.4
	c-1.5-1.6-4-1.7-5.6-0.2c-0.1,0.1-0.1,0.1-0.2,0.2L75.6,50.6V36c0-2-1.4-3.7-3.4-4.1h-0.7c-2.3,0-4.1,1.8-4.1,4.1v21.8
	c0,4.1,3.3,7.4,7.4,7.4L96.2,64.8z M30.2,101.9c1.5,1.6,4,1.6,5.6,0.2c0.1-0.1,0.1-0.1,0.2-0.2l20.5-20.5V96c0,2,1.4,3.7,3.4,4.1
	h0.7c2.3,0,4.1-1.8,4.1-4.1V74.2c0-4.1-3.3-7.4-7.4-7.4H35.8c-2.3,0-4.1,1.8-4.1,4.1s1.8,4.1,4.1,4.1h14.9L30.1,95.8
	c-1.3,1.6-1.3,3.8,0,5.4L30.2,101.9z M33,20.6h66c6.8,0,12.4,5.5,12.4,12.4v66c0,6.8-5.5,12.4-12.4,12.4H33
	c-6.8,0-12.4-5.5-12.4-12.4V33C20.6,26.2,26.2,20.6,33,20.6z M33,12.4c-11.4,0-20.6,9.2-20.6,20.6v66c0,11.4,9.2,20.6,20.6,20.6h66
	c11.4,0,20.6-9.2,20.6-20.6V33c0-11.4-9.2-20.6-20.6-20.6H33z`
    }, null, -1)
  ]));
}
const I0 = /* @__PURE__ */ d(Z, [["render", B]]), I = {
  name: "IconCancelled"
}, D = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 20 20",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function P(t, l, e, o, n, s) {
  return C(), r("svg", D, l[0] || (l[0] = [
    i("path", {
      d: "M0.84 7.95C0.69 8.61 0.62 9.3 0.62 10C0.62 10.69 0.69 11.38 0.84 12.04C0.96 12.59 1.14 13.12 1.36 13.64C1.58 14.17 1.85 14.67 2.16 15.14C2.51 15.67 2.91 16.17 3.37 16.62C3.82 17.08 4.32 17.48 4.85 17.83C5.32 18.14 5.82 18.41 6.35 18.63C6.87 18.85 7.4 19.03 7.95 19.15C8.61 19.3 9.3 19.37 10 19.37C10.69 19.37 11.38 19.3 12.04 19.15C12.59 19.03 13.12 18.85 13.64 18.63C14.17 18.41 14.67 18.14 15.14 17.83C15.67 17.48 16.17 17.08 16.62 16.62C17.08 16.17 17.48 15.67 17.83 15.14C18.14 14.67 18.41 14.17 18.63 13.64C18.85 13.12 19.03 12.59 19.15 12.04C19.3 11.38 19.37 10.69 19.37 10C19.37 9.3 19.3 8.61 19.15 7.95C19.03 7.4 18.85 6.87 18.63 6.35C18.41 5.82 18.14 5.32 17.83 4.85C17.48 4.32 17.08 3.82 16.62 3.37C16.17 2.91 15.67 2.51 15.14 2.16C14.67 1.85 14.17 1.58 13.64 1.36C13.12 1.14 12.59 0.96 12.04 0.84C11.38 0.69 10.69 0.62 10 0.62C9.3 0.62 8.61 0.69 7.95 0.84C7.4 0.96 6.87 1.14 6.35 1.36C5.82 1.58 5.32 1.85 4.85 2.16C4.32 2.51 3.82 2.91 3.37 3.37C2.91 3.82 2.51 4.32 2.16 4.85C1.85 5.32 1.58 5.82 1.36 6.35C1.14 6.87 0.96 7.4 0.84 7.95ZM2.07 11.79C1.94 11.21 1.87 10.61 1.87 10C1.87 9.38 1.94 8.78 2.07 8.2C2.17 7.73 2.32 7.28 2.51 6.83C2.7 6.38 2.93 5.95 3.19 5.55C3.5 5.08 3.85 4.65 4.25 4.25C4.65 3.85 5.08 3.5 5.55 3.19C5.95 2.93 6.38 2.7 6.83 2.51C7.28 2.32 7.73 2.17 8.2 2.07C8.78 1.94 9.38 1.87 10 1.87C10.61 1.87 11.21 1.94 11.79 2.07C12.26 2.17 12.71 2.32 13.16 2.51C13.61 2.7 14.04 2.93 14.44 3.19C14.91 3.5 15.34 3.85 15.74 4.25C16.14 4.65 16.49 5.08 16.8 5.55C17.06 5.95 17.29 6.38 17.48 6.83C17.67 7.28 17.82 7.73 17.92 8.2C18.05 8.78 18.12 9.38 18.12 10C18.12 10.61 18.05 11.21 17.92 11.79C17.82 12.26 17.67 12.71 17.48 13.16C17.29 13.61 17.06 14.04 16.8 14.44C16.49 14.91 16.14 15.34 15.74 15.74C15.34 16.14 14.91 16.49 14.44 16.8C14.04 17.06 13.61 17.29 13.16 17.48C12.71 17.67 12.26 17.82 11.79 17.92C11.21 18.05 10.61 18.12 10 18.12C9.38 18.12 8.78 18.05 8.2 17.92C7.73 17.82 7.28 17.67 6.83 17.48C6.38 17.29 5.95 17.06 5.55 16.8C5.08 16.49 4.65 16.14 4.25 15.74C3.85 15.34 3.5 14.91 3.19 14.44C2.93 14.04 2.7 13.61 2.51 13.16C2.32 12.71 2.17 12.26 2.07 11.79Z",
      fill: "currentColor",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1),
    i("path", {
      id: "合并",
      d: "M13.2812 13.8755C13.1992 13.8755 13.1235 13.8613 13.0537 13.832C12.9834 13.8032 12.9194 13.7598 12.8613 13.7017L10 10.8403L7.13867 13.7017C7.08057 13.7598 7.0166 13.8032 6.94629 13.832C6.87646 13.8613 6.80078 13.8755 6.71875 13.8755C6.63721 13.8755 6.56152 13.8613 6.4917 13.832C6.42139 13.8032 6.35742 13.7598 6.29883 13.7017C6.24072 13.6431 6.19678 13.5786 6.16797 13.5083C6.13916 13.4385 6.125 13.3633 6.125 13.2817C6.125 13.2002 6.13916 13.125 6.16797 13.0552C6.19678 12.9849 6.24072 12.9204 6.29883 12.8618L9.16064 10.0005L6.29883 7.13916C6.24072 7.08105 6.19678 7.0166 6.16797 6.9458C6.13916 6.87646 6.125 6.80078 6.125 6.71973C6.125 6.63818 6.13916 6.5625 6.16797 6.49316L6.16797 6.49316C6.19678 6.42236 6.24072 6.35791 6.29883 6.2998C6.35742 6.24121 6.42188 6.19775 6.49219 6.16846C6.56201 6.14014 6.6377 6.12549 6.71875 6.12549C6.80029 6.12549 6.87598 6.14014 6.94531 6.16846C7.01562 6.19775 7.08008 6.24121 7.13867 6.2998L10 9.16113L12.8613 6.2998C13.0015 6.15967 13.1411 6.08984 13.2812 6.08984C13.4209 6.08984 13.561 6.15967 13.7012 6.2998C13.7593 6.35791 13.8032 6.42236 13.832 6.49316C13.8608 6.5625 13.875 6.63818 13.875 6.71973C13.875 6.80078 13.8608 6.87646 13.832 6.9458L13.832 6.9458C13.8032 7.0166 13.7593 7.08105 13.7012 7.13916L10.8394 10.0005L13.7012 12.8618C13.7593 12.9204 13.8032 12.9849 13.832 13.0552C13.8608 13.125 13.875 13.2002 13.875 13.2817C13.875 13.3633 13.8608 13.4385 13.832 13.5083C13.8032 13.5786 13.7593 13.6431 13.7012 13.7017C13.6426 13.7598 13.5786 13.8032 13.5083 13.832C13.4385 13.8613 13.3628 13.8755 13.2812 13.8755Z",
      "clip-rule": "evenodd",
      fill: "currentColor",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1)
  ]));
}
const D0 = /* @__PURE__ */ d(I, [["render", P]]), j = {
  name: "IconClose"
}, E = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 24 24",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function S(t, l, e, o, n, s) {
  return C(), r("svg", E, l[0] || (l[0] = [
    i("defs", null, null, -1),
    i("path", {
      id: "路径 2",
      d: "M4.5 4.5L19.5 19.5",
      stroke: "currentColor",
      "stroke-opacity": "1.000000",
      "stroke-width": "1.500000",
      "stroke-linejoin": "round",
      "stroke-linecap": "round"
    }, null, -1),
    i("path", {
      id: "路径 2",
      d: "M19.5 4.5L4.5 19.5",
      stroke: "currentColor",
      "stroke-opacity": "1.000000",
      "stroke-width": "1.500000",
      "stroke-linejoin": "round",
      "stroke-linecap": "round"
    }, null, -1)
  ]));
}
const P0 = /* @__PURE__ */ d(j, [["render", S]]), z = {
  name: "IconCopy"
}, b = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 16 16",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function V(t, l, e, o, n, s) {
  return C(), r("svg", b, l[0] || (l[0] = [
    i("defs", null, null, -1),
    i("path", {
      id: "矢量 22",
      d: "M4 4.76C2.89 4.76 2 5.66 2 6.76L2 11.99C2 13.1 2.89 13.99 4 13.99L9.23 13.99C10.33 13.99 11.23 13.1 11.23 11.99L11.23 6.76C11.23 5.66 10.33 4.76 9.23 4.76L4 4.76ZM14 8.46L14 3.84C14 2.82 13.17 2 12.15 2L7.53 2",
      stroke: "#191919",
      "stroke-opacity": "1.000000",
      "stroke-width": "1.000000",
      "stroke-linecap": "round"
    }, null, -1)
  ]));
}
const j0 = /* @__PURE__ */ d(z, [["render", V]]), A = {
  name: "IconDelete"
}, U = {
  width: "1em",
  height: "1em",
  fill: "currentColor",
  viewBox: "0 0 16 16",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function G(t, l, e, o, n, s) {
  return C(), r("svg", U, l[0] || (l[0] = [
    i("desc", null, " Created with Pixso. ", -1),
    i("defs", null, null, -1),
    i("path", {
      id: "形状",
      d: "M11.1274 3.62L14.1301 3.62C14.3801 3.62 14.6304 3.87012 14.6304 4.12036C14.6304 4.37061 14.3801 4.62085 14.1301 4.62085L1.86963 4.62085C1.61938 4.62085 1.36914 4.37061 1.36914 4.12036C1.36914 3.87012 1.61938 3.62 1.86963 3.62L4.78882 3.62L5.45605 1.86841C5.6228 1.36804 6.12329 1.03442 6.62354 1.03442L9.37598 1.03442C9.95972 1.03442 10.4602 1.36804 10.627 1.86841L11.1274 3.62ZM6.3728 2.12048C6.3728 2.03711 6.53979 1.95374 6.70654 1.95374L9.37549 1.95374C9.54224 1.95374 9.62573 2.03711 9.70898 2.12048L10.1262 3.53845L5.87256 3.53845L6.3728 2.12048ZM12.0452 13.129L12.0452 5.78943C12.0452 5.53918 12.2954 5.28894 12.5457 5.28894C12.7959 5.28894 13.0461 5.53918 13.0461 5.78943L13.0461 13.129C13.0461 14.1299 12.2119 14.9639 11.2112 14.9639L4.45532 14.9639C3.45459 14.9639 2.62061 14.1299 2.62061 13.129L2.62061 5.78943C2.62061 5.53918 2.87061 5.28894 3.12085 5.28894C3.37109 5.28894 3.62134 5.53918 3.62134 5.78943L3.62134 13.129C3.62134 13.546 3.95508 13.963 4.45532 13.963L11.2112 13.963C11.6282 13.963 12.0452 13.6294 12.0452 13.129Z",
      "clip-rule": "evenodd",
      fill: "#191919",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1),
    i("path", {
      id: "路径",
      d: "M5.78 11.87L5.78 7.29C5.78 7.04 6.04 6.79 6.29 6.79C6.54 6.79 6.79 7.04 6.79 7.29L6.79 11.87C6.79 12.04 6.7 12.12 6.62 12.21C6.54 12.29 6.37 12.37 6.29 12.37C5.95 12.37 5.78 12.12 5.78 11.87Z",
      fill: "#191919",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1),
    i("path", {
      id: "路径_1_",
      d: "M8.87 11.87L8.87 7.29C8.87 7.04 9.12 6.79 9.37 6.79C9.62 6.79 9.87 7.04 9.87 7.29L9.87 11.87C9.87 12.12 9.62 12.37 9.37 12.37C9.12 12.37 8.87 12.12 8.87 11.87Z",
      fill: "#191919",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1)
  ]));
}
const E0 = /* @__PURE__ */ d(A, [["render", G]]), O = {
  name: "IconDislike"
}, H = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 16 16",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function R(t, l, e, o, n, s) {
  return C(), r("svg", H, l[0] || (l[0] = [
    i("defs", null, [
      i("clipPath", { id: "clip597_81528" }, [
        i("rect", {
          id: "推荐-recommended",
          width: "16.000000",
          height: "16.000000",
          transform: "matrix(1 0 0 -1 0 16)",
          fill: "white",
          "fill-opacity": "0"
        })
      ])
    ], -1),
    i("g", { "clip-path": "url(#clip597_81528)" }, [
      i("path", {
        id: "c",
        d: "M2.85 1.33C2.03 1.33 1.35 1.98 1.33 2.78L1.33 2.9L1.33 7.88C1.33 8.71 1.98 9.4 2.78 9.45L2.89 9.45L4.46 9.45L5.46 10.37C5.89 10.76 6.19 11.27 6.32 11.83L6.35 11.99L6.64 13.57C6.78 14.41 7.69 14.87 8.46 14.52C9.44 14.08 10.03 13.06 9.92 11.99L9.91 11.84L9.71 10.53C9.71 10.53 9.3 8.85 9.14 8.77C8.94 8.67 8.87 8.45 8.62 8.5C8.37 8.53 8.05 8.77 8.21 9.17L8.57 10.13C8.57 10.13 8.71 10.69 8.71 10.74L8.91 12C9.01 12.68 8.64 13.35 8.01 13.62C7.87 13.69 7.69 13.61 7.62 13.45L7.3 11.73C7.17 10.98 6.8 10.28 6.26 9.76L6.12 9.63L5.14 8.74L5.14 2.33L11.8 2.33C12.28 2.33 12.71 2.65 12.83 3.11L12.85 3.21L13.62 7.13C13.76 7.79 13.28 8.41 12.64 8.45L8.62 8.5C8.33 8.5 8.14 8.67 8.14 8.95C8.14 9.2 8.33 9.41 8.57 9.45L12.6 9.45C13.87 9.45 14.83 8.32 14.62 7.06L14.6 6.93L13.83 3.01C13.64 2.08 12.85 1.38 11.91 1.33L11.8 1.33L2.85 1.33ZM4.14 8.45L2.89 8.45C2.6 8.45 2.37 8.24 2.33 7.95L2.33 2.9C2.33 2.6 2.55 2.36 2.82 2.33L4.16 2.33L4.14 8.45Z",
        fill: "#191919",
        "fill-opacity": "1.000000",
        "fill-rule": "nonzero"
      })
    ], -1)
  ]));
}
const S0 = /* @__PURE__ */ d(O, [["render", R]]), N = {
  name: "IconEditPen"
}, T = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 16 16",
  fill: "currentColor",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function W(t, l, e, o, n, s) {
  return C(), r("svg", T, l[0] || (l[0] = [
    i("desc", null, " Created with Pixso. ", -1),
    i("defs", null, null, -1),
    i("path", {
      id: "矢量 36",
      d: "M4.64 13.86C4.83 13.78 4.99 13.67 5.13 13.53L12.82 5.84C12.95 5.71 13.07 5.57 13.16 5.42C13.24 5.29 13.31 5.15 13.37 5.01C13.43 4.87 13.47 4.72 13.51 4.57C13.54 4.41 13.56 4.24 13.56 4.07C13.56 3.9 13.54 3.73 13.51 3.57C13.47 3.42 13.43 3.27 13.37 3.13C13.31 2.99 13.24 2.85 13.16 2.72C13.07 2.57 12.95 2.43 12.82 2.3C12.69 2.17 12.55 2.06 12.4 1.96C12.27 1.88 12.14 1.81 12 1.75C11.85 1.7 11.71 1.65 11.56 1.62C11.39 1.59 11.23 1.57 11.06 1.57C10.88 1.57 10.72 1.59 10.55 1.62C10.4 1.65 10.26 1.7 10.11 1.75C9.97 1.81 9.84 1.88 9.71 1.96C9.56 2.06 9.42 2.17 9.29 2.3L1.6 9.99C1.45 10.14 1.35 10.3 1.27 10.48C1.19 10.66 1.16 10.86 1.16 11.06L1.16 13.27C1.16 13.37 1.17 13.46 1.21 13.54C1.24 13.62 1.29 13.7 1.36 13.76C1.43 13.83 1.51 13.88 1.59 13.92C1.67 13.95 1.76 13.97 1.86 13.97L4.07 13.97C4.27 13.97 4.46 13.93 4.64 13.86ZM12.12 5.13L10.99 6.26L8.87 4.13L9.99 3.01C10.08 2.93 10.17 2.86 10.26 2.8C10.33 2.75 10.41 2.71 10.49 2.68C10.57 2.65 10.66 2.62 10.74 2.6C10.85 2.58 10.95 2.57 11.06 2.57C11.16 2.57 11.27 2.58 11.37 2.6C11.45 2.62 11.54 2.65 11.62 2.68C11.7 2.71 11.78 2.75 11.85 2.8C11.94 2.86 12.03 2.93 12.12 3.01C12.2 3.09 12.27 3.18 12.33 3.28C12.37 3.35 12.41 3.43 12.45 3.51C12.48 3.59 12.51 3.67 12.52 3.76C12.54 3.86 12.56 3.96 12.56 4.07C12.56 4.18 12.54 4.28 12.52 4.38C12.51 4.47 12.48 4.55 12.45 4.63C12.37 4.82 12.26 4.99 12.12 5.13ZM4.42 12.82L10.28 6.96L8.16 4.84L2.3 10.7C2.2 10.8 2.16 10.92 2.16 11.06L2.16 12.97L4.07 12.97C4.21 12.97 4.33 12.92 4.42 12.82ZM7.17 13.45C7.17 13.17 7.39 12.95 7.67 12.95L12.67 12.95C12.95 12.95 13.17 13.17 13.17 13.45C13.17 13.73 12.95 13.95 12.67 13.95L7.67 13.95C7.39 13.95 7.17 13.73 7.17 13.45Z",
      fill: "#191919",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1)
  ]));
}
const z0 = /* @__PURE__ */ d(N, [["render", W]]), J = {
  name: "IconEdit"
}, Q = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 16 16",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function X(t, l, e, o, n, s) {
  return C(), r("svg", Q, l[0] || (l[0] = [
    h('<defs></defs><rect id="矩形" width="16.000000" height="16.000000" fill="#FF8800" fill-opacity="0"></rect><rect id="矩形" x="0.500000" y="0.500000" width="15.000000" height="15.000000" stroke="#979797" stroke-opacity="0" stroke-width="1.000000"></rect><path id="路径" d="M4.57 14L12.71 5.85C13.42 5.14 13.42 3.99 12.71 3.28C12 2.57 10.85 2.57 10.14 3.28L2 11.42L2 14L4.57 14Z" stroke="#1476ff" stroke-opacity="1.000000" stroke-width="1.000000" stroke-linejoin="round"></path><path id="Fill 1" d="M12.85 12.75C12.27 13.29 12 14.14 12 15C12 14.15 11.7 13.29 11.14 12.75C10.6 12.22 9.81 12 9 12C9.78 12 10.6 11.73 11.14 11.25C11.74 10.7 12 9.87 12 9C12 9.87 12.26 10.7 12.85 11.25C13.39 11.74 14.21 12 15 12C14.2 12 13.39 12.24 12.85 12.75Z" fill="#1476ff" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="Fill 1" d="" fill="#F2F2F2" fill-opacity="0" fill-rule="evenodd"></path><path id="Fill 1备份" d="M3.57 3.5C3.18 3.86 3 4.42 3 5C3 4.43 2.8 3.86 2.42 3.5C2.06 3.15 1.54 3 1 3C1.52 3 2.07 2.82 2.42 2.5C2.82 2.13 3 1.58 3 1C3 1.58 3.17 2.13 3.57 2.5C3.92 2.82 4.47 3 5 3C4.46 3 3.92 3.16 3.57 3.5Z" fill="#1476ff" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="Fill 1备份" d="" fill="#F2F2F2" fill-opacity="0" fill-rule="evenodd"></path><path id="路径" d="M8.85 4.57L11.42 7.14" stroke="#1476ff" stroke-opacity="1.000000" stroke-width="1.000000" stroke-linejoin="round" stroke-linecap="round"></path>', 9)
  ]));
}
const b0 = /* @__PURE__ */ d(J, [["render", X]]), Y = {
  name: "IconEmptySearch"
}, $ = {
  width: "7em",
  height: "7em",
  viewBox: "0 0 112 112",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function q(t, l, e, o, n, s) {
  return C(), r("svg", $, l[0] || (l[0] = [
    h('<defs></defs><path id="矩形备份-2" d="M77.18 14L88.67 14C91.82 14 94.37 16.55 94.37 19.7L94.37 32.17C94.37 32.96 93.73 33.6 92.95 33.6L77.18 33.6L77.18 14Z" fill="#EEEEEE" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="矩形备份-2" d="M94.3774 19.6016C94.3774 16.5088 91.8123 14.0016 88.6477 14.0016L77.1885 14.0016L77.1885 33.6016L92.9451 33.6016C93.7361 33.6016 94.3774 32.9748 94.3774 32.2016L94.3774 19.6016ZM78.4785 15.2609L88.6475 15.2609C91.0991 15.2609 93.0867 17.2034 93.0867 19.5996L93.0867 32.1996C93.0867 32.2762 93.0229 32.3383 92.9446 32.3383L78.4785 32.3383L78.4785 15.2609Z" clip-rule="evenodd" fill="#818181" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="矩形" d="M20.33 13.99L88.62 13.99L88.62 13.99C88.88 14.79 89 15.19 89 15.19C82.79 15.19 83.59 19.05 83.59 31C83.59 37.03 83.59 41.87 83.59 45.5C83.59 45.5 83.59 45.5 83.59 45.5L83.59 94.62C83.59 98.03 80.83 100.8 77.41 100.8L20.33 100.8C16.92 100.8 14.16 98.03 14.16 94.62L14.16 20.17C14.16 16.76 16.92 13.99 20.33 13.99Z" fill="#FFFFFF" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="矩形" d="M20.3701 13.9999L88.624 13.9999L89.01 15.1953L88.6494 15.1998C83.4832 15.3343 83.5078 18.4558 83.5784 27.3533C83.5864 28.3734 83.595 29.4695 83.5964 30.6457L83.5967 94.7332C83.5967 98.0837 80.8176 100.8 77.3896 100.8L20.3701 100.8C16.9421 100.8 14.1631 98.0837 14.1631 94.7332L14.1631 20.0665C14.1631 16.716 16.9421 13.9999 20.3701 13.9999ZM15.4531 20.0665C15.4531 17.4126 17.6543 15.2611 20.3699 15.2611L84.3958 15.2628C83.7549 15.819 83.281 16.5607 82.9536 17.5081C82.47 18.9081 82.3086 20.2378 82.2808 23.616L82.2769 25.3833L82.3035 29.7987L82.3059 94.7332C82.3059 97.3871 80.1045 99.5386 77.3892 99.5386L20.3699 99.5386C17.6543 99.5386 15.4531 97.3871 15.4531 94.7332L15.4531 20.0665Z" clip-rule="evenodd" fill="#818181" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="路径备份-3" d="M77.7 100.8C79.61 100.8 58.99 100.8 15.86 100.8C10.96 100.8 7 96.83 7 91.93L7 84.01C7 83.49 7.42 83.06 7.95 83.06L70.63 83.06C71.16 83.06 71.58 83.49 71.58 84.01C71.58 87.49 71.58 90.96 71.58 94.44C71.58 96.72 73.17 99.55 76.55 99.55C78.68 99.55 76.67 100.8 77.7 100.8Z" fill="#FFFFFF" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="路径备份-3" d="M70.6318 83.0671C71.1594 83.0671 71.5869 83.485 71.5869 84.0005L71.5869 94.4444C71.5869 96.6768 73.1042 99.436 76.3428 99.5524L76.5564 99.5562C77.8093 99.5562 77.626 99.9882 77.4756 100.343C77.3706 100.59 77.2815 100.8 77.7061 100.8L16.0715 100.8C11.0613 100.8 6.99951 96.8307 6.99951 91.9338L6.99951 84.0005C6.99951 83.485 7.427 83.0671 7.95435 83.0671L70.6318 83.0671ZM8.29053 84.3267L70.2966 84.3267L70.2969 94.4427L70.2991 94.6143C70.3472 96.5544 71.2087 98.3509 72.6643 99.5056L72.7053 99.5375L16.072 99.5375L15.8772 99.5352C11.6697 99.4342 8.29053 96.0688 8.29053 91.9321L8.29053 84.3267Z" clip-rule="evenodd" fill="#818181" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="Fill-1" d="M34.07 22.19C27.83 22.19 22.75 27.16 22.75 33.25C22.75 39.35 27.83 44.32 34.07 44.32C40.3 44.32 45.38 39.35 45.38 33.25C45.38 27.16 40.3 22.19 34.07 22.19Z" fill="#1476FF" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="形状结合" d="M38.31 28.84C38.71 29.24 38.65 29.93 38.18 30.39L35.57 32.95L38.18 35.51C38.63 35.94 38.7 36.58 38.37 36.99L38.31 37.06C37.9 37.46 37.19 37.4 36.72 36.94L34.1 34.38L31.49 36.94C31.01 37.4 30.3 37.46 29.9 37.06C29.49 36.66 29.55 35.97 30.02 35.51L32.64 32.95L30.02 30.39C29.57 29.96 29.5 29.32 29.83 28.91L29.9 28.84C30.3 28.44 31.01 28.5 31.49 28.96L34.1 31.52L36.72 28.96C37.19 28.5 37.9 28.44 38.31 28.84Z" fill="#FFFFFF" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="形状结合" d="M38.3096 28.8433C38.7144 29.239 38.6594 29.9346 38.1875 30.3961L35.5718 32.9534L38.1875 35.5113C38.6301 35.944 38.7058 36.5825 38.3801 36.9866L38.3096 37.0641C37.905 37.4597 37.1936 37.4062 36.7214 36.9447L34.1057 34.3871L31.4895 36.9447C31.0176 37.4062 30.3059 37.4597 29.9014 37.0641C29.4968 36.6685 29.5515 35.9729 30.0237 35.5113L32.6389 32.9534L30.0237 30.3961C29.5811 29.9635 29.5051 29.325 29.8308 28.9209L29.9014 28.8434C30.3059 28.4477 31.0176 28.5013 31.4895 28.9628L34.1057 31.5197L36.7214 28.9628C37.1936 28.5013 37.905 28.4477 38.3096 28.8433ZM30.4766 29.3516C30.5288 29.3024 30.7583 29.3196 30.9463 29.5035L34.106 32.5916L37.2651 29.5035C37.4531 29.3196 37.6829 29.3023 37.7664 29.384C37.8457 29.4616 37.8289 29.6747 37.6443 29.8554L34.4756 32.9533L37.644 36.052C37.8184 36.2222 37.8445 36.4279 37.7759 36.5128L37.7351 36.5559C37.6829 36.6051 37.4531 36.5878 37.2651 36.4039L34.106 33.3149L30.9463 36.4039C30.7583 36.5878 30.5286 36.6051 30.4451 36.5234C30.3657 36.4458 30.3826 36.2327 30.5674 36.052L33.7356 32.9533L30.5674 29.8554C30.3933 29.6852 30.3672 29.4795 30.4355 29.3947L30.4766 29.3516Z" clip-rule="evenodd" fill="#FFFFFF" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="Fill-1" d="M91.98 78.09L88.4 74.59C88.09 74.29 88.09 73.8 88.4 73.5C88.7 73.2 89.2 73.2 89.51 73.5L93.09 77.01C93.4 77.31 93.4 77.79 93.09 78.09C92.94 78.24 92.74 78.32 92.54 78.32C92.34 78.32 92.14 78.24 91.98 78.09Z" fill="#818181" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="Fill-1" d="M89.51 73.5L93.09 77.01C93.4 77.31 93.4 77.79 93.09 78.09C92.94 78.24 92.74 78.32 92.54 78.32C92.34 78.32 92.14 78.24 91.98 78.09L88.4 74.59C88.09 74.29 88.09 73.8 88.4 73.5C88.7 73.2 89.2 73.2 89.51 73.5Z" fill="#818181" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="矩形" d="M92.3589 77.3622C93.3684 76.3616 94.9844 76.3504 95.98 77.3371L103.173 84.4643C104.123 85.4058 104.152 86.9293 103.263 87.9314L103.148 88.0529C102.139 89.0535 100.522 89.0647 99.5269 88.078L92.334 80.9508C91.3838 80.0092 91.3545 78.4858 92.2437 77.4837L92.3589 77.3622ZM93.4229 78.4483L93.4292 78.4419C93.8369 78.0378 94.4895 78.0333 94.8916 78.4318L102.084 85.559C102.471 85.9422 102.47 86.5707 102.081 86.9629L101.996 87.0405C101.591 87.3725 100.991 87.3544 100.612 86.9794L93.4194 79.8522C93.0327 79.4689 93.0342 78.8404 93.4229 78.4483Z" clip-rule="evenodd" fill="#818181" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="矩形" d="M96.01 77.3232C95.0259 76.3391 93.4119 76.3575 92.405 77.3645L92.3042 77.4707C91.3979 78.4819 91.4136 80.0193 92.3638 80.9695L99.4912 88.0972C100.475 89.0813 102.09 89.0628 103.096 88.0559L103.197 87.9496C104.104 86.9384 104.088 85.401 103.138 84.4508L96.01 77.3232ZM93.2761 80.0538C92.7842 79.5617 92.7932 78.7547 93.2969 78.2513C93.8003 77.7478 94.6072 77.7386 95.0994 78.2306L102.227 85.3583C102.719 85.8503 102.71 86.6573 102.206 87.1608C101.703 87.6643 100.896 87.6735 100.404 87.1814L93.2761 80.0538Z" clip-rule="evenodd" fill="#818181" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="路径" d="M77.32 43.41C87.27 43.41 95.37 51.33 95.37 61.05C95.37 70.78 87.27 78.69 77.32 78.69C67.37 78.69 59.27 70.78 59.27 61.05C59.27 51.33 67.37 43.41 77.32 43.41Z" fill="#FFFFFF" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="路径" d="M95.3748 61.0587C95.3748 51.3318 87.2783 43.4187 77.3262 43.4187C67.3743 43.4187 59.2778 51.3318 59.2778 61.0587C59.2778 70.7856 67.3743 78.6987 77.3262 78.6987C87.2783 78.6987 95.3748 70.7856 95.3748 61.0587ZM60.8206 61.0573C60.8206 52.1745 68.2224 44.9401 77.325 44.9401C86.4275 44.9401 93.8293 52.1745 93.8293 61.0573C93.8293 69.94 86.4275 77.1745 77.325 77.1745C68.2224 77.1745 60.8206 69.94 60.8206 61.0573Z" clip-rule="evenodd" fill="#818181" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="路径" d="M95.3748 61.0587C95.3748 51.3318 87.2783 43.4187 77.3262 43.4187C67.3743 43.4187 59.2778 51.3318 59.2778 61.0587C59.2778 70.7856 67.3743 78.6987 77.3262 78.6987C87.2783 78.6987 95.3748 70.7856 95.3748 61.0587ZM60.5659 61.0587C60.5659 52.0277 68.085 44.6787 77.3252 44.6787C86.5652 44.6787 94.0845 52.0277 94.0845 61.0587C94.0845 70.0897 86.5652 77.4387 77.3252 77.4387C68.085 77.4387 60.5659 70.0897 60.5659 61.0587Z" clip-rule="evenodd" fill="#818181" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="路径" d="M95.3748 61.0587C95.3748 51.3318 87.2783 43.4187 77.3262 43.4187L77.0283 43.4211C67.2131 43.5771 59.2778 51.4291 59.2778 61.0587C59.2778 70.7856 67.3743 78.6987 77.3262 78.6987L77.6243 78.6963C87.4395 78.5403 95.3748 70.6883 95.3748 61.0587ZM60.5659 61.0587C60.5659 52.0277 68.085 44.6787 77.3252 44.6787C86.5652 44.6787 94.0845 52.0277 94.0845 61.0587C94.0845 70.0897 86.5652 77.4387 77.3252 77.4387C68.085 77.4387 60.5659 70.0897 60.5659 61.0587Z" clip-rule="evenodd" fill="#818181" fill-opacity="1.000000" fill-rule="evenodd"></path><circle id="椭圆形备份" cx="33.599854" cy="33.599899" r="11.200000" fill="#1576FF" fill-opacity="1.000000"></circle><path id="路径-20" d="M38.12 37C38.42 37.3 38.43 37.78 38.13 38.09C37.83 38.39 37.34 38.39 37.04 38.09L29.13 30.26C28.83 29.96 28.83 29.47 29.13 29.17C29.43 28.87 29.92 28.87 30.22 29.17L38.12 37Z" fill="#FFFFFF" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="路径-20" d="M38.09 29.13C38.39 29.43 38.39 29.92 38.09 30.22L30.26 38.12C29.96 38.42 29.47 38.43 29.17 38.13C28.87 37.83 28.86 37.34 29.16 37.04L37 29.13C37.3 28.83 37.78 28.83 38.09 29.13Z" fill="#FFFFFF" fill-opacity="1.000000" fill-rule="evenodd"></path>', 21)
  ]));
}
const V0 = /* @__PURE__ */ d(Y, [["render", q]]), K = {
  name: "IconError"
}, l1 = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 20 20",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function i1(t, l, e, o, n, s) {
  return C(), r("svg", l1, l[0] || (l[0] = [
    i("path", {
      id: "路径 (边框)",
      d: "M0.84 7.95C0.69 8.61 0.62 9.3 0.62 10C0.62 10.69 0.69 11.38 0.84 12.04C0.96 12.59 1.14 13.12 1.36 13.64C1.58 14.17 1.85 14.67 2.16 15.14C2.51 15.67 2.91 16.17 3.37 16.62C3.82 17.08 4.32 17.48 4.85 17.83C5.32 18.14 5.82 18.41 6.35 18.63C6.87 18.85 7.4 19.03 7.95 19.15C8.61 19.3 9.3 19.37 10 19.37C10.69 19.37 11.38 19.3 12.04 19.15C12.59 19.03 13.12 18.85 13.64 18.63C14.17 18.41 14.67 18.14 15.14 17.83C15.67 17.48 16.17 17.08 16.62 16.62C17.08 16.17 17.48 15.67 17.83 15.14C18.14 14.67 18.41 14.17 18.63 13.64C18.85 13.12 19.03 12.59 19.15 12.04C19.3 11.38 19.37 10.69 19.37 10C19.37 9.3 19.3 8.61 19.15 7.95C19.03 7.4 18.85 6.87 18.63 6.35C18.41 5.82 18.14 5.32 17.83 4.85C17.48 4.32 17.08 3.82 16.62 3.37C16.17 2.91 15.67 2.51 15.14 2.16C14.67 1.85 14.17 1.58 13.64 1.36C13.12 1.14 12.59 0.96 12.04 0.84C11.38 0.69 10.69 0.62 10 0.62C8.72 0.62 7.51 0.87 6.35 1.36C5.82 1.58 5.32 1.85 4.85 2.16C4.32 2.51 3.82 2.91 3.37 3.37C2.91 3.82 2.51 4.32 2.16 4.85C1.85 5.32 1.58 5.82 1.36 6.35C1.14 6.87 0.96 7.4 0.84 7.95ZM2.07 11.79C1.94 11.21 1.87 10.61 1.87 10C1.87 9.38 1.94 8.78 2.07 8.2C2.17 7.73 2.32 7.28 2.51 6.83C2.7 6.38 2.93 5.95 3.19 5.55C3.5 5.08 3.85 4.65 4.25 4.25C4.65 3.85 5.08 3.5 5.55 3.19C5.95 2.93 6.38 2.7 6.83 2.51C7.28 2.32 7.73 2.17 8.2 2.07C8.78 1.94 9.38 1.87 10 1.87C10.61 1.87 11.21 1.94 11.79 2.07C12.26 2.17 12.71 2.32 13.16 2.51C13.61 2.7 14.04 2.93 14.44 3.19C14.91 3.5 15.34 3.85 15.74 4.25C16.14 4.65 16.49 5.08 16.8 5.55C17.06 5.95 17.29 6.38 17.48 6.83C17.67 7.28 17.82 7.73 17.92 8.2C18.05 8.78 18.12 9.38 18.12 10C18.12 10.61 18.05 11.21 17.92 11.79C17.82 12.26 17.67 12.71 17.48 13.16C17.07 14.13 16.49 14.99 15.74 15.74C14.99 16.49 14.13 17.07 13.16 17.48C12.71 17.67 12.26 17.82 11.79 17.92C11.21 18.05 10.61 18.12 10 18.12C9.38 18.12 8.78 18.05 8.2 17.92C7.73 17.82 7.28 17.67 6.83 17.48C5.86 17.07 5 16.49 4.25 15.74C3.85 15.34 3.5 14.91 3.19 14.44C2.93 14.04 2.7 13.61 2.51 13.16C2.32 12.71 2.17 12.26 2.07 11.79Z",
      fill: "currentColor",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1),
    i("path", {
      id: "合并",
      d: "M10 5.5C9.65479 5.5 9.375 5.77979 9.375 6.125L9.375 10.875C9.375 11.2202 9.65479 11.5 10 11.5C10.3452 11.5 10.625 11.2202 10.625 10.875L10.625 6.125C10.625 5.77979 10.3452 5.5 10 5.5ZM9.0625 13.5127C9.0625 14.0303 9.48242 14.4502 10 14.4502C10.5176 14.4502 10.9375 14.0303 10.9375 13.5127C10.9375 12.9951 10.5176 12.5752 10 12.5752C9.48242 12.5752 9.0625 12.9951 9.0625 13.5127Z",
      "clip-rule": "evenodd",
      fill: "currentColor",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd",
      mask: "url(#mask_922_195863)"
    }, null, -1)
  ]));
}
const A0 = /* @__PURE__ */ d(K, [["render", i1]]), t1 = {
  name: "IconFileExcel"
}, e1 = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 20 20",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function o1(t, l, e, o, n, s) {
  return C(), r("svg", e1, l[0] || (l[0] = [
    i("path", {
      id: "矩形备份-77",
      d: "M12.91 1.66L17.29 6.04L13.86 6.04C13.34 6.04 12.91 5.61 12.91 5.09L12.91 1.66L12.91 1.66Z",
      fill: "#7EEDC1",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1),
    i("path", {
      id: "矩形备份-23",
      d: "M12.97 1.66L12.97 5.08C12.96 5.61 13.39 6.04 13.92 6.04L17.34 6.04L17.34 16.66C17.34 17.58 16.59 18.33 15.67 18.33L4.42 18.33C3.5 18.33 2.76 17.58 2.76 16.66L2.76 3.33C2.76 2.41 3.5 1.66 4.42 1.66L12.97 1.66L12.97 1.66Z",
      fill: "#06BB73",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1),
    i("path", {
      id: "路径-5",
      d: "M8.22 9.21L11.73 14.21",
      stroke: "#FFFFFF",
      "stroke-opacity": "1.000000",
      "stroke-width": "0.722222",
      "stroke-linejoin": "round",
      "stroke-linecap": "round"
    }, null, -1),
    i("path", {
      id: "路径-5",
      d: "M11.79 9.23L8.29 14.24",
      stroke: "#FFFFFF",
      "stroke-opacity": "1.000000",
      "stroke-width": "0.722222",
      "stroke-linejoin": "round",
      "stroke-linecap": "round"
    }, null, -1)
  ]));
}
const U0 = /* @__PURE__ */ d(t1, [["render", o1]]), n1 = {
  name: "IconFileFolder"
}, r1 = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 20 20",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function C1(t, l, e, o, n, s) {
  return C(), r("svg", r1, l[0] || (l[0] = [
    i("path", {
      id: "形状结合",
      d: "M1.66 4.58C1.66 3.66 2.41 2.91 3.33 2.91L9.28 2.91C9.81 2.91 10.3 3.16 10.61 3.58L11.05 4.16L16.66 4.16C17.58 4.16 18.33 4.91 18.33 5.83L18.33 15C18.33 15.92 17.58 16.66 16.66 16.66L3.33 16.66C2.41 16.66 1.66 15.92 1.66 15L1.66 4.58Z",
      fill: "#FFB321",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1),
    i("path", {
      id: "path-1",
      d: "M10.61 3.58L11.66 4.98L11.66 4.98L11.66 7.5L1.66 7.5L1.66 4.58C1.66 3.66 2.41 2.91 3.33 2.91L9.28 2.91C9.81 2.91 10.3 3.16 10.61 3.58Z",
      fill: "#F4A407",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1),
    i("path", {
      id: "矩形",
      d: "M3.33 4.16L16.66 4.16C17.12 4.16 17.49 4.54 17.49 5L17.49 15C17.49 15.46 17.12 15.83 16.66 15.83L3.33 15.83C2.87 15.83 2.49 15.46 2.49 15L2.49 5C2.49 4.54 2.87 4.16 3.33 4.16Z",
      fill: "#FFFFFF",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1),
    i("path", {
      id: "矩形",
      d: "M3.33 5.41L6.67 5.41C7.07 5.41 7.45 5.27 7.75 5.01L8.28 4.56C8.58 4.3 8.96 4.16 9.36 4.16L10.83 4.16L10.83 4.16L16.66 4.16C17.58 4.16 18.33 4.91 18.33 5.83L18.33 15C18.33 15.92 17.58 16.66 16.66 16.66L3.33 16.66C2.41 16.66 1.66 15.92 1.66 15L1.66 7.08C1.66 6.16 2.41 5.41 3.33 5.41Z",
      fill: "#FFB219",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1)
  ]));
}
const G0 = /* @__PURE__ */ d(n1, [["render", C1]]), d1 = {
  name: "IconFileImage"
}, s1 = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 20 20",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function h1(t, l, e, o, n, s) {
  return C(), r("svg", s1, l[0] || (l[0] = [
    h('<g opacity="0.000000"><rect id="Rectangle" width="20.000000" height="20.000000" fill="#000000" fill-opacity="1.000000"></rect></g><path id="Path" d="M17.61 6.14L17.61 16.66C17.61 17.71 16.76 18.57 15.71 18.57L4.28 18.57C3.23 18.57 2.38 17.71 2.38 16.66L2.38 3.33C2.38 2.28 3.23 1.42 4.28 1.42L12.9 1.42L17.61 6.14Z" fill="#FF8800" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="路径" d="M12.63 9.87C12.73 9.95 12.82 10.03 12.89 10.13L14.81 12.64C15.06 12.97 14.83 13.44 14.41 13.44L5.72 13.44C5.31 13.44 5.07 12.97 5.32 12.64L6.25 11.43C6.48 11.13 6.92 11.07 7.22 11.3C7.27 11.34 7.31 11.38 7.35 11.43L7.73 11.93C8.13 12.45 8.92 12.45 9.32 11.93L10.69 10.13C11.15 9.52 12.02 9.41 12.63 9.87Z" fill="#FFFFFF" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="路径" d="" fill="#979797" fill-opacity="0" fill-rule="evenodd"></path><circle id="椭圆形" cx="8.214355" cy="8.928375" r="1.071429" fill="#FFFFFF" fill-opacity="1.000000"></circle><circle id="椭圆形" cx="8.214355" cy="8.928375" r="0.571429" stroke="#979797" stroke-opacity="0" stroke-width="1.000000"></circle><path id="Path" d="M12.9 5.19L12.9 1.42L17.61 6.14L13.85 6.14C13.33 6.14 12.9 5.71 12.9 5.19L12.9 5.19Z" fill="#FFC693" fill-opacity="1.000000" fill-rule="evenodd"></path>', 7)
  ]));
}
const O0 = /* @__PURE__ */ d(d1, [["render", h1]]), c1 = {
  name: "IconFileOther"
}, p1 = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 20 20",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function a1(t, l, e, o, n, s) {
  return C(), r("svg", p1, l[0] || (l[0] = [
    h('<path id="矩形备份-77" d="M12.91 1.66L17.29 6.04L13.86 6.04C13.34 6.04 12.91 5.61 12.91 5.08L12.91 1.66L12.91 1.66Z" fill="#DBDBDB" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="矩形备份-23" d="M12.97 1.66L12.97 5.08C12.96 5.61 13.39 6.04 13.92 6.04L17.34 6.04L17.34 16.66C17.34 17.58 16.59 18.33 15.67 18.33L4.42 18.33C3.5 18.33 2.76 17.58 2.76 16.66L2.76 3.33C2.76 2.41 3.5 1.66 4.42 1.66L12.97 1.66L12.97 1.66Z" fill="#C2C2C2" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="多边形 1" d="M6.8 8.95L6.8 12.7L10.05 14.58L13.29 12.7L13.29 8.95L10.05 7.08L6.8 8.95Z" stroke="#FFFFFF" stroke-opacity="1.000000" stroke-width="0.666667" stroke-linejoin="round"></path><path id="多边形 2" d="M6.8 8.75L10.05 10.62L13.29 8.75" stroke="#FFFFFF" stroke-opacity="1.000000" stroke-width="0.666667" stroke-linejoin="round"></path><path id="矢量 3" d="M10.05 10.75L10.05 14.58" stroke="#FFFFFF" stroke-opacity="1.000000" stroke-width="0.666667" stroke-linejoin="round" stroke-linecap="round"></path>', 5)
  ]));
}
const H0 = /* @__PURE__ */ d(c1, [["render", a1]]), w1 = {
  name: "IconFilePdf"
}, L1 = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 20 20",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function f1(t, l, e, o, n, s) {
  return C(), r("svg", L1, l[0] || (l[0] = [
    h('<path id="矩形备份-47" d="M12.91 1.66L17.29 6.04L13.86 6.04C13.34 6.04 12.91 5.61 12.91 5.09L12.91 1.66L12.91 1.66Z" fill="#FFBFBE" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="矩形备份-23" d="M12.97 1.66L12.97 5.08C12.96 5.61 13.39 6.04 13.92 6.04L17.34 6.04L17.34 16.66C17.34 17.58 16.59 18.33 15.67 18.33L4.42 18.33C3.5 18.33 2.76 17.58 2.76 16.66L2.76 3.33C2.76 2.41 3.5 1.66 4.42 1.66L12.97 1.66L12.97 1.66Z" fill="#FF4C48" fill-opacity="1.000000" fill-rule="evenodd"></path><path id="多边形" d="M11.67 13.81L8.38 13.81L10.02 10.96L11.67 13.81Z" stroke="#FFFFFF" stroke-opacity="1.000000" stroke-width="0.541667" stroke-linejoin="round"></path><path id="路径" d="M10.07 10.71C10.48 10.23 10.68 9.87 10.68 9.63C10.68 9.26 10.38 8.95 10.02 8.95C9.66 8.95 9.36 9.26 9.36 9.63C9.36 9.87 9.57 10.23 9.97 10.71" stroke="#FFFFFF" stroke-opacity="1.000000" stroke-width="0.541667" stroke-linejoin="round" stroke-linecap="round"></path><path id="路径备份" d="M11.87 13.96C12.08 14.55 12.29 14.91 12.49 15.03C12.82 15.22 13.23 15.11 13.41 14.8C13.59 14.48 13.48 14.08 13.15 13.89C12.94 13.77 12.53 13.77 11.92 13.88" stroke="#FFFFFF" stroke-opacity="1.000000" stroke-width="0.541667" stroke-linejoin="round" stroke-linecap="round"></path><path id="路径备份-2" d="M8.13 13.88C7.51 13.77 7.1 13.77 6.89 13.89C6.57 14.08 6.45 14.48 6.63 14.8C6.81 15.11 7.22 15.22 7.55 15.03C7.76 14.91 7.97 14.55 8.17 13.96" stroke="#FFFFFF" stroke-opacity="1.000000" stroke-width="0.541667" stroke-linejoin="round" stroke-linecap="round"></path>', 6)
  ]));
}
const R0 = /* @__PURE__ */ d(w1, [["render", f1]]), u1 = {
  name: "IconFilePpt"
}, g1 = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 20 20",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function x1(t, l, e, o, n, s) {
  return C(), r("svg", g1, l[0] || (l[0] = [
    i("path", {
      id: "矩形备份-24",
      d: "M12.91 1.66L17.29 6.04L13.86 6.04C13.34 6.04 12.91 5.61 12.91 5.09L12.91 1.66L12.91 1.66Z",
      fill: "#FEC9B0",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1),
    i("path", {
      id: "矩形备份-23",
      d: "M12.97 1.66L12.97 5.08C12.96 5.61 13.39 6.04 13.92 6.04L17.34 6.04L17.34 16.66C17.34 17.58 16.59 18.33 15.67 18.33L4.42 18.33C3.5 18.33 2.76 17.58 2.76 16.66L2.76 3.33C2.76 2.41 3.5 1.66 4.42 1.66L12.97 1.66L12.97 1.66Z",
      fill: "#FF7737",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1),
    i("path", {
      id: "路径-7",
      d: "M10.27 9.27L8.35 9.27C8.22 9.27 8.12 9.37 8.12 9.5L8.12 14.94",
      stroke: "#FFFFFF",
      "stroke-opacity": "1.000000",
      "stroke-width": "0.767361",
      "stroke-linejoin": "round",
      "stroke-linecap": "round"
    }, null, -1),
    i("path", {
      id: "路径",
      d: "M8.47 12.46L10.29 12.46C11.17 12.46 11.88 11.75 11.88 10.87C11.88 9.99 11.17 9.27 10.29 9.27",
      stroke: "#FFFFFF",
      "stroke-opacity": "1.000000",
      "stroke-width": "0.767361",
      "stroke-linejoin": "round",
      "stroke-linecap": "round"
    }, null, -1)
  ]));
}
const N0 = /* @__PURE__ */ d(u1, [["render", x1]]), k1 = {
  name: "IconFileRemove"
}, F1 = {
  width: "16.000000",
  height: "16.000000",
  viewBox: "0 0 16 16",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function m1(t, l, e, o, n, s) {
  return C(), r("svg", F1, l[0] || (l[0] = [
    h('<path id="路径" d="M8 1C4.13 1 1 4.13 1 8C1 11.86 4.13 15 8 15C11.86 15 15 11.86 15 8C15 4.13 11.86 1 8 1Z" fill="#C2C2C2" fill-opacity="1.000000" fill-rule="evenodd"></path><mask id="mask_971_59183" fill="white"><path id="形状结合备份" d="M10.6249 10.9999C10.5289 10.9999 10.433 10.9631 10.3597 10.89L8 8.53027L5.64026 10.89C5.56702 10.9631 5.47107 10.9999 5.37512 10.9999C5.27905 10.9999 5.18311 10.9631 5.10999 10.89C4.96338 10.7434 4.96338 10.5065 5.10999 10.3597L7.46973 8L5.10999 5.64026C4.96338 5.49353 4.96338 5.25659 5.10999 5.10999C5.25659 4.96338 5.49353 4.96338 5.64026 5.10999L8 7.46973L10.3597 5.10999C10.5065 4.96338 10.7434 4.96338 10.89 5.10999C11.0366 5.25659 11.0366 5.49353 10.89 5.64026L8.53027 8L10.89 10.3597C11.0366 10.5065 11.0366 10.7434 10.89 10.89C10.8169 10.9631 10.7209 10.9999 10.6249 10.9999Z" clip-rule="evenodd" fill="" fill-opacity="1.000000" fill-rule="evenodd"></path></mask><path id="形状结合备份" d="M10.6249 10.9999C10.5289 10.9999 10.433 10.9631 10.3597 10.89L8 8.53027L5.64026 10.89C5.56702 10.9631 5.47107 10.9999 5.37512 10.9999C5.27905 10.9999 5.18311 10.9631 5.10999 10.89C4.96338 10.7434 4.96338 10.5065 5.10999 10.3597L7.46973 8L5.10999 5.64026C4.96338 5.49353 4.96338 5.25659 5.10999 5.10999C5.25659 4.96338 5.49353 4.96338 5.64026 5.10999L8 7.46973L10.3597 5.10999C10.5065 4.96338 10.7434 4.96338 10.89 5.10999C11.0366 5.25659 11.0366 5.49353 10.89 5.64026L8.53027 8L10.89 10.3597C11.0366 10.5065 11.0366 10.7434 10.89 10.89C10.8169 10.9631 10.7209 10.9999 10.6249 10.9999Z" clip-rule="evenodd" fill="#FFFFFF" fill-opacity="1.000000" fill-rule="evenodd" mask="url(#mask_971_59183)"></path><path id="形状结合备份" d="M10.6249 11.0999C10.4938 11.0999 10.3818 11.0535 10.2891 10.9607L7.92932 8.60095L8 8.53027L8.07068 8.60095L5.71094 10.9607C5.61816 11.0535 5.50623 11.0999 5.37512 11.0999C5.24402 11.0999 5.13208 11.0535 5.03931 10.9607C4.94641 10.8679 4.90002 10.756 4.90002 10.6249C4.90002 10.4939 4.94641 10.382 5.03931 10.2891L7.39905 7.92932L7.46973 8L7.39905 8.07068L5.03931 5.71094C4.94641 5.61804 4.90002 5.5061 4.90002 5.37512C4.90002 5.24402 4.94641 5.13208 5.03931 5.03931C5.13208 4.94641 5.24402 4.90002 5.37512 4.90002C5.5061 4.90002 5.61804 4.94641 5.71094 5.03931L8.07068 7.39905L8 7.46973L7.92932 7.39905L10.2891 5.03931C10.382 4.94641 10.4939 4.90002 10.6249 4.90002C10.756 4.90002 10.8679 4.94641 10.9607 5.03931C11.0536 5.13208 11.1 5.24402 11.1 5.37512C11.1 5.5061 11.0536 5.61804 10.9607 5.71094L8.60095 8.07068L8.53027 8L8.60095 7.92932L10.9607 10.2891C11.0536 10.382 11.1 10.4939 11.1 10.6249C11.1 10.756 11.0536 10.8679 10.9607 10.9607C10.8679 11.0535 10.756 11.0999 10.6249 11.0999ZM10.6249 10.8999C10.7008 10.8999 10.7656 10.873 10.8193 10.8193C10.873 10.7655 10.9 10.7007 10.9 10.6249C10.9 10.5491 10.8732 10.4843 10.8193 10.4305L8.38879 8L10.8193 5.56946C10.873 5.51575 10.9 5.45093 10.9 5.37512C10.9 5.29932 10.873 5.2345 10.8193 5.18066C10.7655 5.12683 10.7007 5.09998 10.6249 5.09998C10.5491 5.09998 10.4843 5.12683 10.4305 5.18066L8 7.61121L5.56946 5.18066C5.51575 5.12683 5.45093 5.09998 5.37512 5.09998C5.29932 5.09998 5.2345 5.12683 5.18066 5.18066C5.12683 5.2345 5.09998 5.29932 5.09998 5.37512C5.09998 5.45093 5.12683 5.51575 5.18066 5.56946L7.61121 8L5.18066 10.4305C5.12683 10.4843 5.09998 10.5491 5.09998 10.6249C5.09998 10.7007 5.12683 10.7655 5.18066 10.8193C5.23438 10.873 5.29919 10.8999 5.37512 10.8999C5.45093 10.8999 5.51575 10.873 5.56946 10.8193L8 8.38879L10.4305 10.8193C10.4843 10.873 10.5491 10.8999 10.6249 10.8999Z" fill="#FFFFFF" fill-opacity="1.000000" fill-rule="nonzero"></path>', 4)
  ]));
}
const T0 = /* @__PURE__ */ d(k1, [["render", m1]]), v1 = {
  name: "IconFileWord"
}, y1 = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 20 20",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function M1(t, l, e, o, n, s) {
  return C(), r("svg", y1, l[0] || (l[0] = [
    i("path", {
      id: "矩形备份-6",
      d: "M16.74 5.62L17.16 5.62L17.16 6.04L16.74 6.04L16.74 5.62Z",
      fill: "#FFFFFF",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1),
    i("path", {
      id: "矩形备份-23",
      d: "M12.97 1.66L12.97 5.08C12.96 5.61 13.39 6.04 13.92 6.04L17.34 6.04L17.34 16.66C17.34 17.58 16.59 18.33 15.67 18.33L4.42 18.33C3.5 18.33 2.76 17.58 2.76 16.66L2.76 3.33C2.76 2.41 3.5 1.66 4.42 1.66L12.97 1.66L12.97 1.66Z",
      fill: "#3B8CFA",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1),
    i("path", {
      id: "矩形备份-24",
      d: "M12.91 1.66L17.29 6.04L13.86 6.04C13.34 6.04 12.91 5.61 12.91 5.09L12.91 1.66L12.91 1.66Z",
      fill: "#ADCDF9",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1),
    i("path", {
      id: "路径-4",
      d: "M7.39 10L8.45 14.24C8.48 14.34 8.61 14.34 8.65 14.25L9.92 10.29C9.95 10.2 10.08 10.2 10.11 10.29L11.37 14.24C11.4 14.34 11.54 14.33 11.57 14.23L12.6 10",
      stroke: "#FFFFFF",
      "stroke-opacity": "1.000000",
      "stroke-width": "0.722222",
      "stroke-linejoin": "round",
      "stroke-linecap": "round"
    }, null, -1)
  ]));
}
const W0 = /* @__PURE__ */ d(v1, [["render", M1]]), Z1 = {
  name: "IconFullScreen"
}, _1 = {
  width: "1em",
  height: "1em",
  version: "1.1",
  id: "图层_1",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink",
  x: "0px",
  y: "0px",
  viewBox: "0 0 132 132",
  style: { "enable-background": "new 0 0 132 132" },
  "xml:space": "preserve"
};
function B1(t, l, e, o, n, s) {
  return C(), r("svg", _1, l[0] || (l[0] = [
    i("rect", {
      style: { fill: "none" },
      width: "132",
      height: "132"
    }, null, -1),
    i("path", {
      style: { fill: "currentColor" },
      d: `M74.2,28.9c-2.3,0-4.1,1.8-4.1,4.1s1.8,4.1,4.1,4.1h14.9L68.6,57.8c-1.5,1.4-1.7,3.7-0.4,5.4l0.4,0.5
	c1.7,1.6,4.2,1.6,5.9,0l20.5-20.5v14.7c0,2,1.4,3.7,3.4,4H99c2.3,0,4.1-1.8,4.1-4.1V36.3c0-4.1-3.3-7.4-7.4-7.4H74.2z M63.4,68.4
	c-1.6-1.6-4.2-1.6-5.9,0L37.1,88.9V74.2c0-2-1.4-3.7-3.4-4H33c-2.3,0-4.1,1.8-4.1,4.1v21.4c0,4.1,3.3,7.4,7.4,7.4h21.5
	c2.3,0,4.1-1.8,4.1-4.1S60.1,95,57.8,95H42.8l20.6-20.7c1.3-1.6,1.3-3.8,0-5.4V68.4z M33,20.6h66c6.8,0,12.4,5.5,12.4,12.4v66
	c0,6.8-5.5,12.4-12.4,12.4H33c-6.8,0-12.4-5.5-12.4-12.4V33C20.6,26.2,26.2,20.6,33,20.6z M33,12.4c-11.4,0-20.6,9.2-20.6,20.6v66
	c0,11.4,9.2,20.6,20.6,20.6h66c11.4,0,20.6-9.2,20.6-20.6V33c0-11.4-9.2-20.6-20.6-20.6H33z`
    }, null, -1)
  ]));
}
const J0 = /* @__PURE__ */ d(Z1, [["render", B1]]), I1 = {
  name: "IconImageLoading"
}, D1 = {
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink",
  width: "20",
  height: "20",
  fill: "none",
  customFrame: "#000000"
};
function P1(t, l, e, o, n, s) {
  return C(), r("svg", D1, l[0] || (l[0] = [
    h('<defs><clipPath id="clipPath_0"><rect width="16" height="16" x="2" y="2" fill="rgb(255,255,255)"></rect></clipPath><linearGradient id="paint_linear_0" x1="21.5809536" x2="6.56828499" y1="9.9996109" y2="2.59961891" gradientUnits="userSpaceOnUse"><stop stop-color="rgb(255,255,255)" offset="0.000956000003" stop-opacity="0"></stop><stop stop-color="rgb(255,255,255)" offset="1" stop-opacity="1"></stop></linearGradient></defs><rect id="附件状态" width="20" height="20" x="0" y="0"></rect><rect id="矩形" width="20" height="20" x="0" y="0" fill="rgb(216,216,216)" fill-opacity="0"></rect><rect id="矩形" width="18" height="18" x="1" y="1" stroke="rgb(151,151,151)" stroke-opacity="0" stroke-width="2"></rect>', 4),
    i("g", {
      id: "编组备份 21",
      "clip-path": "url(#clipPath_0)",
      customFrame: "url(#clipPath_0)"
    }, [
      i("rect", {
        id: "编组备份 21",
        width: "16",
        height: "16",
        x: "2",
        y: "2"
      }),
      i("path", {
        id: "Fill 1",
        d: "M2 18L18 18L18 2L2 2L2 18Z",
        fill: "rgb(229,229,229)",
        "fill-opacity": "0",
        "fill-rule": "evenodd"
      }),
      i("path", {
        id: "路径",
        d: "M10.0002 2.59961C10.4952 2.59961 10.8952 2.99961 10.8952 3.49961C10.8952 3.99461 10.4952 4.39461 10.0002 4.39461C6.90522 4.39461 4.40022 6.90461 4.40022 9.99961C4.40022 13.0896 6.90522 15.5996 10.0002 15.5996C13.0902 15.5996 15.6002 13.0896 15.6002 9.99961C15.6002 9.49961 16.0002 9.09961 16.5002 9.09961C16.9952 9.09961 17.4002 9.49961 17.4002 9.99961C17.4002 14.0846 14.0852 17.3996 10.0002 17.3996C5.91021 17.3996 2.59521 14.0846 2.59521 9.99961C2.59521 5.90961 5.91021 2.59961 10.0002 2.59961Z",
        fill: "url(#paint_linear_0)",
        "fill-rule": "evenodd"
      })
    ], -1),
    i("animateTransform", {
      attributeName: "transform",
      type: "rotate",
      from: "0 0 0",
      to: "360 0 0",
      dur: "1s",
      repeatCount: "indefinite"
    }, null, -1)
  ]));
}
const Q0 = /* @__PURE__ */ d(I1, [["render", P1]]), j1 = {
  name: "IconImageWarning"
}, E1 = {
  viewBox: "0 0 16 16",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink",
  width: "16",
  height: "16",
  fill: "none"
};
function S1(t, l, e, o, n, s) {
  return C(), r("svg", E1, l[0] || (l[0] = [
    h('<rect id="提示" width="16" height="16" x="0" y="0"></rect><rect id="矩形" width="16" height="16" x="0" y="0" fill="rgb(216,216,216)" fill-opacity="0"></rect><rect id="矩形" width="14.999999" height="15" x="0.5" y="0.5" fill="rgb(0,0,0)" fill-opacity="0"></rect><rect id="矩形" width="14.999999" height="15" x="0.5" y="0.5" stroke="rgb(151,151,151)" stroke-opacity="0" stroke-width="1"></rect><path id="路径" d="M8 1C4.13 1 1 4.13 1 8C1 11.86 4.13 15 8 15C11.86 15 15 11.86 15 8C15 4.13 11.86 1 8 1Z" fill="rgb(255,136,2)" fill-opacity="0" fill-rule="evenodd"></path><path id="路径" d="M1 8C1 11.86 4.13 15 8 15C11.86 15 15 11.86 15 8C15 4.13 11.86 1 8 1C4.13 1 1 4.13 1 8Z" fill-rule="evenodd" stroke="rgb(255,254.745,254.745)" stroke-width="1"></path><path id="路径" d="M8.00014 5.71C7.52014 5.71 7.14014 5.33 7.14014 4.85C7.14014 4.38 7.52014 4 8.00014 4C8.47014 4 8.85014 4.38 8.85014 4.85C8.85014 5.33 8.47014 5.71 8.00014 5.71Z" fill="rgb(255,255,255)" fill-rule="evenodd"></path><rect id="矩形" width="1.1428566" height="5.14285707" x="7.42822266" y="6.85742188" rx="0.071428299" fill="rgb(255,255,255)"></rect>', 8)
  ]));
}
const X0 = /* @__PURE__ */ d(j1, [["render", S1]]), z1 = {
  name: "IconLike"
}, b1 = {
  width: "16",
  height: "16",
  viewBox: "0 0 16 16",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function V1(t, l, e, o, n, s) {
  return C(), r("svg", b1, l[0] || (l[0] = [
    i("defs", null, [
      i("clipPath", { id: "clip597_81525" }, [
        i("rect", {
          id: "推荐-recommended",
          width: "16.000000",
          height: "16.000000",
          fill: "white",
          "fill-opacity": "0"
        })
      ])
    ], -1),
    i("g", { "clip-path": "url(#clip597_81525)" }, [
      i("path", {
        id: "c",
        d: "M2.85 14.66C2.03 14.66 1.35 14.01 1.33 13.21L1.33 13.09L1.33 8.11C1.33 7.28 1.98 6.59 2.78 6.54L2.89 6.54L4.46 6.54L5.46 5.62C5.89 5.23 6.19 4.72 6.32 4.16L6.35 4L6.64 2.42C6.78 1.58 7.69 1.12 8.46 1.47C9.44 1.91 10.03 2.93 9.92 4L9.91 4.15L9.71 5.46C9.71 5.46 9.3 7.14 9.14 7.22C8.94 7.32 8.87 7.54 8.62 7.5C8.37 7.46 8.05 7.22 8.21 6.82L8.57 5.86C8.57 5.86 8.71 5.3 8.71 5.25L8.91 4C9.01 3.31 8.64 2.64 8.01 2.37C7.87 2.3 7.69 2.38 7.62 2.54L7.3 4.26C7.17 5.01 6.8 5.71 6.26 6.23L6.12 6.36L5.14 7.25L5.14 13.66L11.8 13.66C12.28 13.66 12.71 13.34 12.83 12.88L12.85 12.78L13.62 8.86C13.76 8.2 13.28 7.58 12.64 7.54L8.62 7.5C8.33 7.5 8.14 7.32 8.14 7.04C8.14 6.79 8.33 6.58 8.57 6.54L12.6 6.54C13.87 6.54 14.83 7.67 14.62 8.93L14.6 9.06L13.83 12.98C13.64 13.91 12.85 14.61 11.91 14.66L11.8 14.66L2.85 14.66ZM4.14 7.54L2.89 7.54C2.6 7.54 2.37 7.75 2.33 8.04L2.33 13.09C2.33 13.39 2.55 13.63 2.82 13.66L4.16 13.66L4.14 7.54Z",
        fill: "#191919",
        "fill-opacity": "1.000000",
        "fill-rule": "nonzero"
      })
    ], -1)
  ]));
}
const Y0 = /* @__PURE__ */ d(z1, [["render", V1]]), A1 = {
  name: "IconLoadingSpeech"
}, U1 = {
  width: "152.343506",
  height: "18.000000",
  viewBox: "0 0 152.344 18",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function G1(t, l, e, o, n, s) {
  return C(), r("svg", U1, l[0] || (l[0] = [
    h('<defs><linearGradient x1="159.655945" y1="6.532532" x2="-2.217805" y2="3.665466" id="paint_linear_498_7491_0" gradientUnits="userSpaceOnUse"><stop stop-color="#E6FF7D" stop-opacity="0.200000"></stop><stop offset="0.264015" stop-color="#FFB22E" stop-opacity="0.200000"></stop><stop offset="0.522782" stop-color="#97A7FF" stop-opacity="0.258824"></stop><stop offset="0.702223" stop-color="#76ADFF" stop-opacity="0.400000"></stop><stop offset="0.830214" stop-color="#67D1FF" stop-opacity="0.301961"></stop><stop offset="1.000000" stop-color="#D27EFF" stop-opacity="0.090196"></stop></linearGradient></defs><path id="Fill 3" d="M9.52 9.61C12.69 9.9 15.86 10.31 19.04 10.74C20.63 10.94 22.21 11.28 23.8 11.78C25.39 12.26 26.97 12.84 28.56 13.26C30.15 13.67 31.73 13.89 33.32 13.89C34.11 13.88 34.91 13.69 35.7 13.33C36.49 12.98 37.29 12.49 38.08 12.06C38.87 11.63 39.67 11.29 40.46 11.19C40.86 11.13 41.25 11.13 41.65 11.17C42.05 11.22 42.45 11.31 42.84 11.39C44.43 11.75 46.02 12.36 47.6 13.19C50.78 14.81 53.95 16.29 57.12 17.28C60.3 18.29 63.47 18.26 66.65 16.96C69.82 15.68 72.99 14.04 76.17 12.73C79.34 11.4 82.51 11.19 85.69 11.85C88.86 12.53 92.04 14.04 95.21 14.41C96 14.51 96.8 14.56 97.59 14.56C98.38 14.55 99.18 14.5 99.97 14.46C101.56 14.35 103.14 14.18 104.73 13.94C107.91 13.45 111.08 12.64 114.25 11.93C117.43 11.19 120.6 10.62 123.77 10.26C126.95 9.88 130.12 9.49 133.3 9.28C136.47 9.06 139.64 9 142.82 9L152.34 9L142.82 9C139.64 8.99 136.47 8.93 133.3 8.71C130.12 8.5 126.95 8.11 123.77 7.73C120.6 7.37 117.43 6.8 114.25 6.06C111.08 5.35 107.91 4.54 104.73 4.05C103.14 3.81 101.56 3.64 99.97 3.53C99.18 3.49 98.38 3.44 97.59 3.43C96.8 3.43 96 3.48 95.21 3.58C92.04 3.95 88.86 5.46 85.69 6.14C82.51 6.8 79.34 6.59 76.17 5.26C72.99 3.95 69.82 2.31 66.65 1.03C63.47 -0.27 60.3 -0.31 57.12 0.71C53.95 1.7 50.78 3.18 47.6 4.8C46.02 5.63 44.43 6.24 42.84 6.6C42.45 6.68 42.05 6.77 41.65 6.82C41.25 6.86 40.86 6.86 40.46 6.8C39.67 6.7 38.87 6.36 38.08 5.93C37.29 5.51 36.49 5.01 35.7 4.66C34.91 4.3 34.11 4.11 33.32 4.1C31.73 4.1 30.15 4.32 28.56 4.73C26.97 5.15 25.39 5.73 23.8 6.21C22.21 6.71 20.63 7.05 19.04 7.25C15.86 7.68 12.69 8.09 9.52 8.38C6.34 8.69 3.17 8.94 0 9C3.17 9.05 6.34 9.3 9.52 9.61Z" fill="url(#paint_linear_498_7491_0)" fill-opacity="1.000000" fill-rule="evenodd"></path>', 2)
  ]));
}
const $0 = /* @__PURE__ */ d(A1, [["render", G1]]), O1 = {
  name: "IconLoading"
}, H1 = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 20 20",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function R1(t, l, e, o, n, s) {
  return C(), r("svg", H1, l[0] || (l[0] = [
    i("defs", null, [
      i("linearGradient", {
        x1: "20.954775",
        y1: "9.999996",
        x2: "6.753603",
        y2: "3.000000",
        id: "paint_linear_928_16009_0",
        gradientUnits: "userSpaceOnUse"
      }, [
        i("stop", {
          offset: "0.000956",
          "stop-color": "currentColor",
          "stop-opacity": "0.000000"
        }),
        i("stop", {
          offset: "1.000000",
          "stop-color": "currentColor"
        })
      ])
    ], -1),
    i("path", {
      id: "Stroke 2",
      d: "M10 3C6.13 3 3 6.13 3 10C3 13.86 6.13 17 10 17C13.86 17 17 13.86 17 10",
      stroke: "url(#paint_linear_928_16009_0)",
      "stroke-opacity": "1.000000",
      "stroke-width": "1.800000",
      "stroke-linejoin": "round",
      "stroke-linecap": "round"
    }, null, -1)
  ]));
}
const q0 = /* @__PURE__ */ d(O1, [["render", R1]]), N1 = {
  name: "IconMenu"
}, T1 = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 20 20",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function W1(t, l, e, o, n, s) {
  return C(), r("svg", T1, l[0] || (l[0] = [
    h('<defs><clipPath id="clip545_56691"><rect id="更多-more" width="20.000000" height="20.000000" fill="white" fill-opacity="0"></rect></clipPath></defs><rect id="组件集 2" width="20.000000" height="20.000000" fill="#FFFFFF" fill-opacity="0"></rect><g clip-path="url(#clip545_56691)"><path id="形状结合" d="M10 1.25C14.83 1.25 18.75 5.16 18.75 10C18.75 14.83 14.83 18.75 10 18.75C5.16 18.75 1.25 14.83 1.25 10C1.25 5.16 5.16 1.25 10 1.25ZM10 2.5C5.85 2.5 2.5 5.85 2.5 10C2.5 14.14 5.85 17.5 10 17.5C14.14 17.5 17.5 14.14 17.5 10C17.5 5.85 14.14 2.5 10 2.5ZM6.06 8.87C6.68 8.87 7.18 9.37 7.18 10C7.18 10.62 6.68 11.12 6.06 11.12C5.44 11.12 4.93 10.62 4.93 10C4.93 9.37 5.44 8.87 6.06 8.87ZM10 8.87C10.62 8.87 11.12 9.37 11.12 10C11.12 10.62 10.62 11.12 10 11.12C9.37 11.12 8.87 10.62 8.87 10C8.87 9.37 9.37 8.87 10 8.87ZM13.93 8.87C14.55 8.87 15.06 9.37 15.06 10C15.06 10.62 14.55 11.12 13.93 11.12C13.31 11.12 12.81 10.62 12.81 10C12.81 9.37 13.31 8.87 13.93 8.87Z" fill="#191919" fill-opacity="1.000000" fill-rule="nonzero"></path></g>', 3)
  ]));
}
const K0 = /* @__PURE__ */ d(N1, [["render", W1]]), J1 = {
  name: "IconPlugin"
}, Q1 = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 20 20",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function X1(t, l, e, o, n, s) {
  return C(), r("svg", Q1, l[0] || (l[0] = [
    h('<defs><clipPath id="clip657_188316"><rect id="接口调用执行 (1)" width="20.000000" height="20.000000" transform="translate(20.000000 0.000000) rotate(90.000000)" fill="white" fill-opacity="0"></rect></clipPath></defs><g clip-path="url(#clip657_188316)"><path id="path" d="M11.87 17.5L11.87 2.53C11.87 2.19 12.15 1.91 12.49 1.91C12.84 1.91 13.12 2.19 13.12 2.53L13.12 17.5C13.12 17.84 12.84 18.12 12.49 18.12C12.15 18.12 11.87 17.84 11.87 17.5Z" fill="currentColor" fill-opacity="1.000000" fill-rule="nonzero"></path><path id="path" d="M3.76 9.99C3.76 9.16 3.92 8.36 4.25 7.61C4.56 6.89 5.01 6.23 5.58 5.67C6.75 4.53 8.3 3.89 9.96 3.89L12.2 3.89L12.2 5.14L9.96 5.14C7.23 5.14 5.01 7.32 5.01 9.99C5.01 12.66 7.23 14.83 9.96 14.83L12.2 14.83L12.2 16.08L9.96 16.08C8.3 16.08 6.75 15.45 5.58 14.3C5.01 13.74 4.56 13.09 4.25 12.36C3.92 11.61 3.76 10.81 3.76 9.99Z" fill="currentColor" fill-opacity="1.000000" fill-rule="nonzero"></path><path id="path" d="M1.25 9.97C1.25 9.63 1.53 9.35 1.87 9.35L4.37 9.35C4.72 9.35 5 9.63 5 9.97C5 10.32 4.72 10.6 4.37 10.6L1.87 10.6C1.53 10.6 1.25 10.32 1.25 9.97ZM12.5 6.25C12.5 5.9 12.78 5.62 13.12 5.62L17.46 5.62C17.8 5.62 18.08 5.9 18.08 6.25C18.08 6.59 17.8 6.87 17.46 6.87L13.12 6.87C12.78 6.87 12.5 6.59 12.5 6.25ZM12.5 13.73C12.5 13.39 12.78 13.11 13.12 13.11L17.46 13.11C17.8 13.11 18.08 13.39 18.08 13.73C18.08 14.08 17.8 14.36 17.46 14.36L13.12 14.36C12.78 14.36 12.5 14.08 12.5 13.73Z" fill="currentColor" fill-opacity="1.000000" fill-rule="nonzero"></path></g>', 2)
  ]));
}
const l3 = /* @__PURE__ */ d(J1, [["render", X1]]), Y1 = {
  name: "IconPlus"
}, $1 = {
  width: "16.000000",
  height: "16.000000",
  viewBox: "0 0 16 16",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function q1(t, l, e, o, n, s) {
  return C(), r("svg", $1, l[0] || (l[0] = [
    i("defs", null, [
      i("clipPath", { id: "clip596_185312" }, [
        i("rect", {
          id: "svg",
          width: "16.000000",
          height: "16.000000",
          fill: "white",
          "fill-opacity": "0"
        })
      ])
    ], -1),
    i("g", { "clip-path": "url(#clip596_185312)" }, [
      i("mask", {
        id: "mask596_185308",
        "mask-type": "alpha",
        maskUnits: "userSpaceOnUse",
        x: "0.000000",
        y: "0.000000",
        width: "16.000000",
        height: "16.000000"
      }, [
        i("rect", {
          id: "创建-create-1",
          width: "16.000000",
          height: "16.000000",
          fill: "#000000",
          "fill-opacity": "1.000000"
        })
      ]),
      i("g", { mask: "url(#mask596_185308)" }, [
        i("path", {
          id: "_合并",
          d: "M7.5 8.5L7.5 13.5C7.5 13.77 7.72 14 8 14C8.26 14 8.5 13.77 8.5 13.5L8.5 8.5L13.5 8.5C13.77 8.5 14 8.27 14 8C14 7.72 13.77 7.5 13.5 7.5L8.5 7.5L8.5 2.5C8.5 2.22 8.26 2 8 2C7.72 2 7.5 2.22 7.5 2.5L7.5 7.5L2.5 7.5C2.22 7.5 2 7.72 2 8C2 8.27 2.22 8.5 2.5 8.5L7.5 8.5Z",
          fill: "#191919",
          "fill-opacity": "1.000000",
          "fill-rule": "evenodd"
        })
      ])
    ], -1)
  ]));
}
const i3 = /* @__PURE__ */ d(Y1, [["render", q1]]), K1 = {
  name: "IconRefresh"
}, l0 = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 16 16",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function i0(t, l, e, o, n, s) {
  return C(), r("svg", l0, l[0] || (l[0] = [
    i("defs", null, [
      i("clipPath", { id: "clip521_64343" }, [
        i("rect", {
          id: "刷新-refresh",
          width: "16.000000",
          height: "16.000000",
          fill: "white",
          "fill-opacity": "0"
        })
      ])
    ], -1),
    i("g", { "clip-path": "url(#clip521_64343)" }, [
      i("path", {
        id: "形状结合",
        d: "M11.8 2.75C14.55 4.73 15.27 8.46 13.54 11.3L15.24 10.84C15.47 10.78 15.7 10.9 15.8 11.1L15.83 11.18C15.9 11.44 15.75 11.7 15.49 11.77L12.59 12.54C12.34 12.61 12.08 12.46 12.01 12.2L11.23 9.31C11.16 9.06 11.31 8.79 11.57 8.72C11.82 8.65 12.08 8.81 12.16 9.06L12.65 10.91C14.21 8.49 13.62 5.24 11.24 3.52C8.77 1.74 5.32 2.28 3.53 4.73C1.74 7.18 2.29 10.61 4.76 12.39C6.42 13.59 8.28 13.84 10.39 13.16C10.64 13.08 10.91 13.22 10.99 13.47C11.07 13.72 10.94 13.99 10.68 14.07C8.29 14.85 6.11 14.55 4.19 13.16C1.3 11.07 0.65 7.04 2.75 4.16C4.85 1.29 8.91 0.66 11.8 2.75Z",
        fill: "#191919",
        "fill-opacity": "1.000000",
        "fill-rule": "nonzero"
      })
    ], -1)
  ]));
}
const t3 = /* @__PURE__ */ d(K1, [["render", i0]]), t0 = {
  name: "IconSearch"
}, e0 = {
  width: "1em",
  height: "1em",
  fill: "currentColor",
  viewBox: "0 0 16 16",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function o0(t, l, e, o, n, s) {
  return C(), r("svg", e0, l[0] || (l[0] = [
    i("desc", null, " Created with Pixso. ", -1),
    i("defs", null, null, -1),
    i("path", {
      id: "矢量 8",
      d: "M1.72 7.17C1.72 4.13 4.18 1.67 7.22 1.67C10.26 1.67 12.72 4.13 12.72 7.17C12.72 8.5 12.25 9.72 11.46 10.67L14.2 13.37C14.39 13.57 14.39 13.88 14.2 14.08C14.03 14.26 13.76 14.28 13.56 14.14L10.76 11.38C9.8 12.19 8.57 12.67 7.22 12.67C4.18 12.67 1.72 10.21 1.72 7.17ZM11.72 7.17C11.72 4.69 9.71 2.67 7.22 2.67C4.74 2.67 2.72 4.69 2.72 7.17C2.72 9.66 4.74 11.67 7.22 11.67C9.71 11.67 11.72 9.66 11.72 7.17Z",
      fill: "#808080",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1)
  ]));
}
const e3 = /* @__PURE__ */ d(t0, [["render", o0]]), n0 = {
  name: "IconSend"
}, r0 = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 36 36",
  fill: "currentColor",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function C0(t, l, e, o, n, s) {
  return C(), r("svg", r0, l[0] || (l[0] = [
    h('<defs><filter id="filter_447_4247_dd" x="0" y="0" width="36" height="36" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB"><feFlood flood-opacity="0" result="BackgroundImageFix"></feFlood><feColorMatrix in="SourceAlpha" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0" result="hardAlpha"></feColorMatrix><feOffset dx="0" dy="2"></feOffset><feGaussianBlur stdDeviation="1.66667"></feGaussianBlur><feComposite in2="hardAlpha" operator="out" k2="-1" k3="1"></feComposite><feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.039 0"></feColorMatrix><feBlend mode="normal" in2="BackgroundImageFix" result="effect_dropShadow_1"></feBlend><feBlend mode="normal" in="SourceGraphic" in2="effect_dropShadow_1" result="shape"></feBlend></filter></defs><g filter="url(#filter_447_4247_dd)"><rect id="Rectangle" x="0" y="0" rx="18" width="36" height="36" fill="currentColor" fill-opacity="1"></rect></g><path id="Fill 1" d="M26.38 8.7L7.99 17.45C7.59 17.64 7.58 18.21 7.98 18.4L12.56 20.7C12.78 20.81 13.05 20.77 13.23 20.62L23.54 11.95C23.71 11.81 23.94 12.05 23.78 12.21L14.9 21.42C14.72 21.61 14.63 21.87 14.66 22.14L15.21 26.9C15.26 27.36 15.83 27.54 16.14 27.19L18.36 24.65C18.49 24.5 18.46 24.25 18.29 24.13L16.45 22.88C16.34 22.81 16.44 22.64 16.56 22.7L19 23.92L21.87 25.35C22.17 25.5 22.52 25.35 22.62 25.03L27.44 9.62C27.64 8.98 26.98 8.41 26.38 8.7Z" fill="#FFFFFF" fill-opacity="1" fill-rule="evenodd"></path>', 3)
  ]));
}
const o3 = /* @__PURE__ */ d(n0, [["render", C0]]), d0 = {
  name: "IconSparkles"
}, s0 = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 14.7168 15.2383",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function h0(t, l, e, o, n, s) {
  return C(), r("svg", s0, l[0] || (l[0] = [
    i("path", {
      id: "形状结合",
      d: "M11.9795 0.21582L11.9863 0.187256C12.0508 -0.0610352 12.4033 -0.0627441 12.4697 0.184814L12.4814 0.226318C12.5918 0.610352 12.9014 0.909912 13.29 1.00757C13.542 1.07104 13.542 1.42896 13.29 1.49243C12.8887 1.59351 12.5723 1.90991 12.4707 2.31104C12.4072 2.56299 12.0498 2.56299 11.9863 2.31104C11.8848 1.90991 11.5684 1.59351 11.168 1.49243C10.916 1.42896 10.916 1.07104 11.168 1.00757C11.5605 0.908691 11.873 0.602783 11.9795 0.21582ZM5.94922 2.06641L5.99316 1.89575C6.12207 1.39917 6.82617 1.39575 6.95996 1.89087L7.03125 2.15552C7.79297 4.79102 9.91504 6.84351 12.5771 7.51392C13.0811 7.64062 13.0811 8.35669 12.5771 8.48364C9.82422 9.17725 7.65625 11.3457 6.96289 14.0989C6.83594 14.6025 6.12012 14.6028 5.99316 14.0989C5.2998 11.3457 3.13086 9.17725 0.37793 8.48413C-0.125977 8.35742 -0.125977 7.64136 0.37793 7.5144C3.07422 6.83496 5.21387 4.73926 5.94922 2.06641ZM12.3301 11.5376L12.3174 11.5835C12.1475 12.2029 11.6475 12.6921 11.0195 12.8506C10.6162 12.9521 10.6162 13.5247 11.0195 13.6262C11.6611 13.7878 12.167 14.2942 12.3291 14.936C12.4307 15.3391 13.0029 15.3391 13.1045 14.936C13.2666 14.2942 13.7725 13.7878 14.415 13.626C14.8174 13.5244 14.8174 12.9519 14.415 12.8503C13.7939 12.6941 13.2988 12.2148 13.1211 11.6003L13.1025 11.5339C12.9961 11.1377 12.4326 11.1406 12.3301 11.5376Z",
      "clip-rule": "evenodd",
      fill: "currentColor",
      "fill-opacity": "1.000000",
      "fill-rule": "evenodd"
    }, null, -1)
  ]));
}
const n3 = /* @__PURE__ */ d(d0, [["render", h0]]), c0 = {
  name: "IconStop"
}, p0 = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 24 24",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function a0(t, l, e, o, n, s) {
  return C(), r("svg", p0, l[0] || (l[0] = [
    h('<defs></defs><rect id="矩形" width="24.000000" height="24.000000" fill="#D8D8D8" fill-opacity="0"></rect><rect id="矩形" x="0.500000" y="0.500000" width="23.000000" height="23.000000" stroke="#979797" stroke-opacity="0" stroke-width="1.000000"></rect><rect id="矩形" x="8.666504" y="8.666626" rx="1.500000" width="6.666667" height="6.666667" fill="currentColor" fill-opacity="1.000000"></rect><circle id="椭圆形" cx="12.000000" cy="12.000000" r="8.250000" fill="#D8D8D8" fill-opacity="0"></circle><circle id="椭圆形" cx="12.000000" cy="12.000000" r="8.250000" stroke="currentColor" stroke-opacity="1.000000" stroke-width="1.500000"></circle><path id="路径" d="M11.8 4.47C11.48 4.39 11.25 4.1 11.25 3.75C11.25 3.33 11.58 3 12 3L12.02 3.02L12.02 3.75C16.56 3.76 20.25 7.45 20.25 12C20.25 16.55 16.55 20.25 12 20.25C10.52 20.25 9.13 19.86 7.93 19.18L7.57 19.81L7.54 19.82C7.18 19.61 7.06 19.16 7.26 18.8C7.44 18.49 7.78 18.35 8.11 18.44L11.8 4.47Z" fill="#595959" fill-opacity="0" fill-rule="evenodd"></path><path id="路径" d="M7.91 19.17C9.12 19.85 10.51 20.25 12 20.25C16.55 20.25 20.25 16.55 20.25 12C20.25 7.44 16.55 3.75 12 3.75" stroke="currentColor" stroke-opacity="1.000000" stroke-width="1.500000" stroke-linecap="round"></path>', 8)
  ]));
}
const r3 = /* @__PURE__ */ d(c0, [["render", a0]]), w0 = {
  name: "IconTypeAll"
}, L0 = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 16 16",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function f0(t, l, e, o, n, s) {
  return C(), r("svg", L0, l[0] || (l[0] = [
    h('<defs></defs><rect id="矩形" width="16.000000" height="16.000000" fill="#6598FF" fill-opacity="0"></rect><rect id="矩形" x="0.500000" y="0.500000" width="15.000000" height="15.000000" stroke="#979797" stroke-opacity="0" stroke-width="1.000000"></rect><rect id="矩形" x="1.667969" y="1.666504" rx="1.000000" width="5.000000" height="5.000000" fill="#1476FF" fill-opacity="0"></rect><rect id="矩形" x="1.667969" y="1.666504" rx="1.000000" width="5.000000" height="5.000000" stroke="#191919" stroke-opacity="1.000000" stroke-width="1.000000"></rect><rect id="矩形备份 6" x="1.500000" y="9.500000" rx="1.000000" width="5.000000" height="5.000000" fill="#1476FF" fill-opacity="0"></rect><rect id="矩形备份 6" x="1.500000" y="9.500000" rx="1.000000" width="5.000000" height="5.000000" stroke="#191919" stroke-opacity="1.000000" stroke-width="1.000000"></rect><rect id="矩形备份 5" x="9.500000" y="1.500000" rx="2.500000" width="5.000000" height="5.000000" fill="#1476FF" fill-opacity="0"></rect><rect id="矩形备份 5" x="9.500000" y="1.500000" rx="2.500000" width="5.000000" height="5.000000" stroke="#191919" stroke-opacity="1.000000" stroke-width="1.000000"></rect><rect id="矩形备份 7" x="9.500000" y="9.500000" rx="1.000000" width="5.000000" height="5.000000" fill="#1476FF" fill-opacity="0"></rect><rect id="矩形备份 7" x="9.500000" y="9.500000" rx="1.000000" width="5.000000" height="5.000000" stroke="#191919" stroke-opacity="1.000000" stroke-width="1.000000"></rect>', 11)
  ]));
}
const C3 = /* @__PURE__ */ d(w0, [["render", f0]]), u0 = {
  name: "IconUpload"
}, g0 = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 20 20",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function x0(t, l, e, o, n, s) {
  return C(), r("svg", g0, l[0] || (l[0] = [
    i("defs", null, [
      i("clipPath", { id: "clip141_56332" }, [
        i("rect", {
          width: "20.000000",
          height: "20.000000",
          fill: "white",
          "fill-opacity": "0"
        })
      ])
    ], -1),
    i("g", { "clip-path": "url(#clip141_56332)" }, [
      i("path", {
        d: "M10.5 5.75C10.5 6.83 11.41 7.75 12.5 7.75C13.58 7.75 14.5 6.83 14.5 5.75C14.5 4.66 13.58 3.75 12.5 3.75C11.41 3.66 10.5 4.58 10.5 5.75ZM13.25 5.75C13.25 6.16 12.91 6.5 12.5 6.5C12.08 6.5 11.75 6.16 11.75 5.75C11.75 5.33 12.08 5 12.5 5C12.91 4.91 13.25 5.25 13.25 5.75ZM12.25 11.25L11.25 12.25C10.66 12.83 9.83 12.83 9.25 12.25L5.58 8.66C5.5 8.58 5.41 8.58 5.33 8.66L2.25 11.75L2.25 13.75C2.25 14.91 3.25 15.83 4.58 15.83L9.75 15.83C10.08 15.83 10.41 16.08 10.41 16.5C10.41 16.83 10.16 17.16 9.75 17.16L4.58 17.16C2.58 17.16 1 15.66 1 13.83L1 4.33C1 2.5 2.58 1 4.58 1L14.91 1C16.91 1 18.5 2.5 18.5 4.33L18.5 9.16C18.5 9.5 18.25 9.83 17.83 9.83C17.5 9.83 17.16 9.58 17.16 9.16L17.16 4.33C17.16 3.16 16.16 2.25 14.83 2.25L4.58 2.25C3.33 2.33 2.33 3.25 2.33 4.33L2.33 10L4.58 7.83C5.16 7.25 6 7.25 6.58 7.83L10.25 11.41C10.33 11.5 10.41 11.5 10.5 11.41L11.5 10.41C11.75 10.16 12.16 10.16 12.41 10.41C12.66 10.66 12.5 11 12.25 11.25ZM19 15C19 15.33 18.75 15.66 18.33 15.66L16.25 15.66L16.25 17.75C16.25 18.08 16 18.41 15.58 18.41C15.25 18.41 14.91 18.16 14.91 17.75L14.91 15.66L12.83 15.66C12.5 15.66 12.16 15.41 12.16 15C12.16 14.58 12.41 14.33 12.83 14.33L15 14.33L15 12.25C15 11.91 15.25 11.58 15.66 11.58C16 11.58 16.33 11.83 16.33 12.25L16.33 14.33L18.33 14.33C18.66 14.33 19 14.58 19 15Z",
        fill: "currentColor",
        "fill-opacity": "1.000000",
        "fill-rule": "nonzero"
      })
    ], -1)
  ]));
}
const d3 = /* @__PURE__ */ d(u0, [["render", x0]]), k0 = {
  name: "IconVoice"
}, F0 = {
  width: "1em",
  height: "1em",
  viewBox: "0 0 32 32",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  "xmlns:xlink": "http://www.w3.org/1999/xlink"
};
function m0(t, l, e, o, n, s) {
  return C(), r("svg", F0, l[0] || (l[0] = [
    h('<defs><linearGradient x1="12.999998" y1="21.128080" x2="20.000002" y2="6.999999" id="paint_linear_447_301_0" gradientUnits="userSpaceOnUse"><stop stop-color="#E280FF"></stop><stop offset="0.517615" stop-color="#6686FF"></stop><stop offset="1.000000" stop-color="#4DDDFF"></stop></linearGradient></defs><rect id="矩形" rx="8.000000" width="32.000000" height="32.000000" fill="#000000" fill-opacity="0"></rect><rect id="矩形" x="0.500000" y="0.500000" rx="8.000000" width="31.000000" height="31.000000" stroke="#979797" stroke-opacity="0" stroke-width="1.000000"></rect><rect id="矩形" x="4.000000" y="4.000000" width="24.000000" height="24.000000" fill="#D8D8D8" fill-opacity="0"></rect><rect id="矩形" x="4.500000" y="4.500000" width="23.000000" height="23.000000" stroke="#979797" stroke-opacity="0" stroke-width="1.000000"></rect><rect id="矩形" x="13.000000" y="7.000000" rx="3.500000" width="7.000000" height="11.000000" fill="url(#paint_linear_447_301_0)" fill-opacity="0"></rect><rect id="矩形" x="13.000000" y="7.000000" rx="3.500000" width="7.000000" height="11.000000" stroke="#595959" stroke-opacity="1.000000" stroke-width="1.200000" stroke-linejoin="round"></rect><path id="路径" d="M8.41 14.02L8.39 14C8.39 13.66 8.66 13.4 9 13.4C9.33 13.4 9.6 13.66 9.6 14L23.39 14C23.39 13.66 23.66 13.4 24 13.4C24.33 13.4 24.6 13.66 24.6 14L24.58 14.02L24 14.02C23.98 18.42 20.63 22 16.5 22C12.36 22 9.01 18.42 9 14.02L8.41 14.02Z" fill="#D8D8D8" fill-opacity="0" fill-rule="evenodd"></path><path id="路径" d="M9 14C9 18.41 12.35 22 16.5 22C20.64 22 24 18.41 24 14" stroke="#595959" stroke-opacity="1.000000" stroke-width="1.200000" stroke-linecap="round"></path><path id="路径 2" d="M15.91 22.6L15.89 22.58C15.89 22.24 16.16 21.98 16.5 21.98C16.83 21.98 17.1 22.24 17.1 22.58L17.08 22.6L15.91 22.6ZM17.08 25.34L17.1 25.36C17.1 25.7 16.83 25.96 16.5 25.96C16.16 25.96 15.89 25.7 15.89 25.36L15.91 25.34L17.08 25.34Z" fill="#D8D8D8" fill-opacity="0" fill-rule="nonzero"></path><path id="路径 2" d="M16.5 22.58L16.5 25.36" stroke="#595959" stroke-opacity="1.000000" stroke-width="1.200000" stroke-linecap="round"></path>', 11)
  ]));
}
const s3 = /* @__PURE__ */ d(k0, [["render", m0]]);
export {
  i3 as $,
  V0 as B,
  Q0 as E,
  A0 as F,
  J0 as G,
  G0 as H,
  $0 as I,
  Y0 as J,
  Z0 as L,
  H0 as M,
  O0 as N,
  j0 as O,
  D0 as P,
  W0 as Q,
  K0 as R,
  R0 as T,
  q0 as V,
  X0 as W,
  E0 as X,
  T0 as Y,
  U0 as Z,
  l3 as _,
  y0 as a,
  S0 as b,
  n3 as d,
  e3 as e,
  _0 as g,
  M0 as h,
  o3 as i,
  N0 as j,
  z0 as k,
  r3 as l,
  I0 as m,
  d3 as p,
  C3 as s,
  t3 as t,
  s3 as u,
  P0 as x,
  b0 as y,
  B0 as z
};
